{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "b63f9d27-093a-475d-a770-564c110d62ea",
   "metadata": {},
   "source": [
    "# GP2 Pathogenic Variant Analysis\n",
    "\n",
    "- **Project:** Parkinsonâ€™s Disease Pathogenic Variants: Cross-Ancestry Analysis and Microarray Data Validation\n",
    "- **Version:** Python/3.10.15\n",
    "    \n",
    "## Notebook Overview\n",
    "\n",
    "- Annotating variants across all ancestries utilizing ANNOVAR's ClinVar database for \"Pathogenic\" variants in established Parkinson's Disease genes (as defined by Blauwendraat et al. 2019)\n",
    "\n",
    "## CHANGELOG\n",
    "\n",
    "- 14-NOV-2024: Notebook started\n",
    "\n",
    "---"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "85a7feb2-e812-4f1d-b20f-ed1476cc7bd9",
   "metadata": {
    "tags": []
   },
   "source": [
    "# Getting Started"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "0397fd8a-9423-41d2-adbd-af32eeb45ed9",
   "metadata": {},
   "source": [
    "## Importing Packages"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "963f7b66-9cbc-4aba-9390-aad22beab3ca",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PACKAGE VERSION (18-NOV-2024)\n",
      "pandas==2.2.3\n",
      "numpy==1.25.2\n"
     ]
    }
   ],
   "source": [
    "import os\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "import sys as sys\n",
    "\n",
    "from datetime import date\n",
    "today = date.today()\n",
    "date = today.strftime(\"%d-%b-%Y\").upper()\n",
    "\n",
    "# print out packages and versions\n",
    "print(f'PACKAGE VERSION ({date})')\n",
    "print('\\n'.join(f'{m.__name__}=={m.__version__}' for m in globals().values() if getattr(m, '__version__', None)))"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "2a8a5cff-6903-43f6-b382-6b6dd17e36eb",
   "metadata": {
    "tags": []
   },
   "source": [
    "# Run Analysis\n",
    "\n",
    "- GP2 release 7 (all ancestries)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "4ba4634c-2929-4aa1-8f36-60de119e6686",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# define paths/variables\n",
    "\n",
    "wd = '' # working directory\n",
    "gp2_tier2 = '' # path to GP2 tier 2 data\n",
    "\n",
    "plink_exec = '' # path to plink executable\n",
    "plink2_exec = '' # path to plink2.0 executable\n",
    "\n",
    "ancestries = ['AAC', 'AFR', 'AJ', 'AMR', 'CAH', 'CAS', 'EAS', 'EUR', 'FIN', 'MDE', 'SAS']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "0d929761",
   "metadata": {},
   "outputs": [],
   "source": [
    "# utility function\n",
    "\n",
    "def shell_do(command):\n",
    "    print(f'Executing: {command}', file=sys.stderr)\n",
    "    !$command"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "31f124a3-38f7-4419-a1e5-5ef0e71b7019",
   "metadata": {},
   "source": [
    "## convert gp2 release 7 pfiles to bfiles"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "85facb67-9ed9-40f2-8da1-6b54364dc49f",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "tags": []
   },
   "outputs": [],
   "source": [
    "# convert the gp2 release 7 pfiles to bfiles for each chromosome of interest for each ancestry group\n",
    "\n",
    "# list of all ancestries\n",
    "ancestries = ['AAC', 'AFR', 'AJ', 'AMR', 'CAH', 'CAS', 'EAS', 'EUR', 'FIN', 'MDE', 'SAS']\n",
    "\n",
    "# list of chromosomes that contain a PD gene of interest\n",
    "chrom_list = ['1','2','3','4','6','12','14','15','16','17','19','20','21','22']\n",
    "\n",
    "for anc in ancestries:\n",
    "    gp2_anc_path = f'{gp2_tier2}/imputed_genotypes/{anc}'\n",
    "    gp2_anc_samples = f'{gp2_tier2}/raw_genotypes/{anc}/{anc}_release7_vwb.samples'\n",
    "    print(f'WORKING ON {anc}')\n",
    "    for chrom in chrom_list:\n",
    "        plink_cmd = f'{plink2_exec} --pfile {gp2_anc_path}/chr{chrom}_{anc}_release7_vwb \\\n",
    "                      --keep {gp2_anc_samples} \\\n",
    "                      --make-bed --out {workspace}/bfiles/{anc}_{chrom}'\n",
    "        shell_do(plink_cmd);\n",
    "    print(f'COMPLETED {anc}')\n",
    "    print('----------------------------------------')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "be9b2fb2-16e9-4096-a970-77566fe40355",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# create merge files for each ancestry\n",
    "\n",
    "for anc in ancestries:\n",
    "    with open(f'{workspace}/bfiles/merge_files_{anc}.txt', 'w') as f:\n",
    "        for chrom in chrom_list:\n",
    "            f.write(f'{workspace}/bfiles/{anc}_{chrom}\\n')\n",
    "    f.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "6fe5739b-e55f-408c-a13e-76bf7516b662",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "tags": []
   },
   "outputs": [],
   "source": [
    "# merge chromosomes together for each ancestry\n",
    "\n",
    "for anc in ancestries:\n",
    "    plink_merge = f'{plink_exec} --merge-list {workspace}/bfiles/merge_files_{anc}.txt \\\n",
    "                    --make-bed --out {workspace}/bfiles/MERGE_{anc}'\n",
    "\n",
    "    shell_do(plink_merge);\n",
    "    print(f'{anc} MERGE COMPLETED')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "160cadd7-12c7-41dd-8439-36816eeb50d1",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# check that all ancestries were merged\n",
    "\n",
    "!ls {workspace}/bfiles/MERGE*"
   ]
  },
  {
   "cell_type": "raw",
   "id": "61247d11-256d-4878-971a-f13f50e9b6ce",
   "metadata": {},
   "source": [
    "to prep files for ANNOVAR annotations:\n",
    "\n",
    "note: check that fid and iid in .fam files are the same\n",
    "note: check that variant IDs have a leading \"chr\" in .bim files\n",
    "note: check that .bim files have no header"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "fdcfce64-a5ff-47d5-bfb2-124851e19604",
   "metadata": {
    "tags": []
   },
   "source": [
    "## annotate the .bim files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "58810630-425d-4016-89ad-cbe42f403a52",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/jupyter/workspace/bfiles\n"
     ]
    }
   ],
   "source": [
    "%cd {workspace}/bfiles"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "e8fb966e-5e02-4577-9ed1-cb0d4a9bb6d6",
   "metadata": {},
   "outputs": [],
   "source": [
    "!cat MERGE_AAC.bim MERGE_AFR.bim MERGE_AJ.bim MERGE_AMR.bim MERGE_CAH.bim MERGE_CAS.bim MERGE_EAS.bim MERGE_EUR.bim MERGE_FIN.bim MERGE_MDE.bim MERGE_SAS.bim | awk '{print $2}' | sort | uniq > all_ancestries.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "d8686a15-f76d-43ea-ad86-a5c58fba70c5",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Processing complete. Output saved to to_annotate.txt\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "# input and output file names\n",
    "input_file=\"all_ancestries.txt\"\n",
    "output_file=\"to_annotate.txt\"\n",
    "\n",
    "# create an empty output file\n",
    "> \"$output_file\"\n",
    "\n",
    "# process each line of the input file\n",
    "while IFS=$':' read -r chr pos ref alt; do\n",
    "    # write the formatted line to the output file\n",
    "    echo \"$chr $pos $pos $ref $alt\" >> \"$output_file\"\n",
    "done < \"$input_file\"\n",
    "\n",
    "echo \"Processing complete. Output saved to $output_file\""
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b25c7b60-3371-42da-81bf-fac9fb5f9c67",
   "metadata": {
    "tags": []
   },
   "source": [
    "### download annovar libraries"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "id": "b2c2251d-b225-4a64-a58f-5132c4b0dc3c",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "NOTICE: Web-based checking to see whether ANNOVAR new version is available ... Done\n",
      "NOTICE: Downloading annotation database http://www.openbioinformatics.org/annovar/download/hg38_refGene.txt.gz ... OK\n",
      "NOTICE: Downloading annotation database http://www.openbioinformatics.org/annovar/download/hg38_refGeneMrna.fa.gz ... OK\n",
      "NOTICE: Downloading annotation database http://www.openbioinformatics.org/annovar/download/hg38_refGeneVersion.txt.gz ... OK\n",
      "NOTICE: Uncompressing downloaded files\n",
      "NOTICE: Finished downloading annotation files for hg38 build version, with files saved at the '/home/jupyter/annovar/humandb' directory\n",
      "NOTICE: Web-based checking to see whether ANNOVAR new version is available ... Done\n",
      "NOTICE: Downloading annotation database http://www.openbioinformatics.org/annovar/download/hg38_avsnp151.txt.gz ... OK\n",
      "NOTICE: Downloading annotation database http://www.openbioinformatics.org/annovar/download/hg38_avsnp151.txt.idx.gz ... OK\n",
      "NOTICE: Uncompressing downloaded files\n",
      "NOTICE: Finished downloading annotation files for hg38 build version, with files saved at the '/home/jupyter/annovar/humandb' directory\n",
      "NOTICE: Web-based checking to see whether ANNOVAR new version is available ... Done\n",
      "NOTICE: Downloading annotation database http://www.openbioinformatics.org/annovar/download/hg38_clinvar_20240917.txt.gz ... OK\n",
      "NOTICE: Downloading annotation database http://www.openbioinformatics.org/annovar/download/hg38_clinvar_20240917.txt.idx.gz ... OK\n",
      "NOTICE: Uncompressing downloaded files\n",
      "NOTICE: Finished downloading annotation files for hg38 build version, with files saved at the '/home/jupyter/annovar/humandb' directory\n",
      "NOTICE: Web-based checking to see whether ANNOVAR new version is available ... Done\n",
      "NOTICE: Downloading annotation database http://www.openbioinformatics.org/annovar/download/hg38_dbnsfp33a.txt.gz ... OK\n",
      "NOTICE: Downloading annotation database http://www.openbioinformatics.org/annovar/download/hg38_dbnsfp33a.txt.idx.gz ... OK\n",
      "NOTICE: Uncompressing downloaded files\n",
      "NOTICE: Finished downloading annotation files for hg38 build version, with files saved at the '/home/jupyter/annovar/humandb' directory\n"
     ]
    }
   ],
   "source": [
    "!/home/jupyter/annovar/annotate_variation.pl -buildver hg38 -downdb -webfrom annovar refGene /home/jupyter/annovar/humandb/\n",
    "!/home/jupyter/annovar/annotate_variation.pl -buildver hg38 -downdb -webfrom annovar avsnp151 /home/jupyter/annovar/humandb/\n",
    "!/home/jupyter/annovar/annotate_variation.pl -buildver hg38 -downdb -webfrom annovar clinvar_20240917 /home/jupyter/annovar/humandb/\n",
    "!/home/jupyter/annovar/annotate_variation.pl -buildver hg38 -downdb -webfrom annovar dbnsfp33a /home/jupyter/annovar/humandb/"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "5a518fb3-784d-4a8e-9e0b-0018d220da56",
   "metadata": {
    "tags": []
   },
   "source": [
    "### run annovar annotations"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d9f25784-913e-437b-a249-1d8551e2d4a2",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "NOTICE: the --polish argument is set ON automatically (use --nopolish to change this behavior)\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Processing operation=g protocol=refGene\n",
      "\n",
      "NOTICE: Running with system command <annotate_variation.pl -geneanno -buildver hg38 -dbtype refGene -outfile all_ancestries_annotated.refGene -exonsort -nofirstcodondel to_annotate.txt /home/jupyter/annovar/humandb/>\n",
      "NOTICE: Output files are written to all_ancestries_annotated.refGene.variant_function, all_ancestries_annotated.refGene.exonic_variant_function\n",
      "NOTICE: Reading gene annotation from /home/jupyter/annovar/humandb/hg38_refGene.txt ... Done with 88819 transcripts (including 21511 without coding sequence annotation) for 28307 unique genes\n",
      "NOTICE: Processing next batch with 5000000 unique variants in 5000000 input lines\n",
      "NOTICE: Finished analyzing 1000000 query variants\n",
      "NOTICE: Finished analyzing 2000000 query variants\n",
      "NOTICE: Finished analyzing 3000000 query variants\n",
      "NOTICE: Finished analyzing 4000000 query variants\n",
      "NOTICE: Reading FASTA sequences from /home/jupyter/annovar/humandb/hg38_refGeneMrna.fa ... Done with 1316 sequences\n",
      "WARNING: A total of 606 sequences will be ignored due to lack of correct ORF annotation\n",
      "NOTICE: Processing next batch with 5000000 unique variants in 5000000 input lines\n",
      "NOTICE: Finished analyzing 1000000 query variants\n",
      "NOTICE: Finished analyzing 2000000 query variants\n",
      "NOTICE: Finished analyzing 3000000 query variants\n",
      "NOTICE: Finished analyzing 4000000 query variants\n",
      "NOTICE: Reading FASTA sequences from /home/jupyter/annovar/humandb/hg38_refGeneMrna.fa ... Done with 1535 sequences\n",
      "WARNING: A total of 606 sequences will be ignored due to lack of correct ORF annotation\n",
      "NOTICE: Processing next batch with 5000000 unique variants in 5000000 input lines\n",
      "NOTICE: Finished analyzing 1000000 query variants\n",
      "NOTICE: Finished analyzing 2000000 query variants\n",
      "NOTICE: Finished analyzing 3000000 query variants\n",
      "NOTICE: Finished analyzing 4000000 query variants\n",
      "NOTICE: Reading FASTA sequences from /home/jupyter/annovar/humandb/hg38_refGeneMrna.fa ... Done with 1364 sequences\n",
      "WARNING: A total of 606 sequences will be ignored due to lack of correct ORF annotation\n",
      "NOTICE: Processing next batch with 5000000 unique variants in 5000000 input lines\n",
      "NOTICE: Finished analyzing 1000000 query variants\n",
      "NOTICE: Finished analyzing 2000000 query variants\n",
      "NOTICE: Finished analyzing 3000000 query variants\n",
      "NOTICE: Finished analyzing 4000000 query variants\n",
      "NOTICE: Reading FASTA sequences from /home/jupyter/annovar/humandb/hg38_refGeneMrna.fa ... Done with 1319 sequences\n",
      "WARNING: A total of 606 sequences will be ignored due to lack of correct ORF annotation\n",
      "NOTICE: Processing next batch with 5000000 unique variants in 5000000 input lines\n",
      "NOTICE: Finished analyzing 1000000 query variants\n",
      "NOTICE: Finished analyzing 2000000 query variants\n",
      "NOTICE: Finished analyzing 3000000 query variants\n",
      "NOTICE: Finished analyzing 4000000 query variants\n",
      "NOTICE: Reading FASTA sequences from /home/jupyter/annovar/humandb/hg38_refGeneMrna.fa ... Done with 1428 sequences\n",
      "WARNING: A total of 606 sequences will be ignored due to lack of correct ORF annotation\n",
      "NOTICE: Processing next batch with 5000000 unique variants in 5000000 input lines\n",
      "NOTICE: Finished analyzing 1000000 query variants\n",
      "NOTICE: Finished analyzing 2000000 query variants\n",
      "NOTICE: Finished analyzing 3000000 query variants\n",
      "NOTICE: Finished analyzing 4000000 query variants\n",
      "NOTICE: Reading FASTA sequences from /home/jupyter/annovar/humandb/hg38_refGeneMrna.fa ... Done with 1636 sequences\n",
      "WARNING: A total of 606 sequences will be ignored due to lack of correct ORF annotation\n",
      "NOTICE: Processing next batch with 5000000 unique variants in 5000000 input lines\n",
      "NOTICE: Finished analyzing 1000000 query variants\n",
      "NOTICE: Finished analyzing 2000000 query variants\n",
      "NOTICE: Finished analyzing 3000000 query variants\n",
      "NOTICE: Finished analyzing 4000000 query variants\n",
      "NOTICE: Reading FASTA sequences from /home/jupyter/annovar/humandb/hg38_refGeneMrna.fa ... Done with 2175 sequences\n",
      "WARNING: A total of 606 sequences will be ignored due to lack of correct ORF annotation\n",
      "NOTICE: Processing next batch with 5000000 unique variants in 5000000 input lines\n",
      "NOTICE: Finished analyzing 1000000 query variants\n",
      "NOTICE: Finished analyzing 2000000 query variants\n",
      "NOTICE: Finished analyzing 3000000 query variants\n",
      "NOTICE: Finished analyzing 4000000 query variants\n",
      "NOTICE: Reading FASTA sequences from /home/jupyter/annovar/humandb/hg38_refGeneMrna.fa ... Done with 2701 sequences\n",
      "WARNING: A total of 606 sequences will be ignored due to lack of correct ORF annotation\n",
      "NOTICE: Processing next batch with 5000000 unique variants in 5000000 input lines\n",
      "NOTICE: Finished analyzing 1000000 query variants\n",
      "NOTICE: Finished analyzing 2000000 query variants\n",
      "NOTICE: Finished analyzing 3000000 query variants\n",
      "NOTICE: Finished analyzing 4000000 query variants\n",
      "NOTICE: Reading FASTA sequences from /home/jupyter/annovar/humandb/hg38_refGeneMrna.fa ... Done with 3385 sequences\n",
      "WARNING: A total of 606 sequences will be ignored due to lack of correct ORF annotation\n",
      "NOTICE: Processing next batch with 5000000 unique variants in 5000000 input lines\n",
      "NOTICE: Finished analyzing 1000000 query variants\n",
      "NOTICE: Finished analyzing 2000000 query variants\n",
      "NOTICE: Finished analyzing 3000000 query variants\n",
      "NOTICE: Finished analyzing 4000000 query variants\n",
      "NOTICE: Reading FASTA sequences from /home/jupyter/annovar/humandb/hg38_refGeneMrna.fa ... Done with 1706 sequences\n",
      "WARNING: A total of 606 sequences will be ignored due to lack of correct ORF annotation\n",
      "NOTICE: Processing next batch with 5000000 unique variants in 5000000 input lines\n",
      "NOTICE: Finished analyzing 1000000 query variants\n",
      "NOTICE: Finished analyzing 2000000 query variants\n",
      "NOTICE: Finished analyzing 3000000 query variants\n",
      "NOTICE: Finished analyzing 4000000 query variants\n",
      "NOTICE: Reading FASTA sequences from /home/jupyter/annovar/humandb/hg38_refGeneMrna.fa ... Done with 1608 sequences\n",
      "WARNING: A total of 606 sequences will be ignored due to lack of correct ORF annotation\n",
      "NOTICE: Processing next batch with 5000000 unique variants in 5000000 input lines\n",
      "NOTICE: Finished analyzing 1000000 query variants\n",
      "NOTICE: Finished analyzing 2000000 query variants\n",
      "NOTICE: Finished analyzing 3000000 query variants\n",
      "NOTICE: Finished analyzing 4000000 query variants\n",
      "NOTICE: Reading FASTA sequences from /home/jupyter/annovar/humandb/hg38_refGeneMrna.fa ... Done with 1168 sequences\n",
      "WARNING: A total of 606 sequences will be ignored due to lack of correct ORF annotation\n",
      "NOTICE: Processing next batch with 5000000 unique variants in 5000000 input lines\n",
      "NOTICE: Finished analyzing 1000000 query variants\n",
      "NOTICE: Finished analyzing 2000000 query variants\n",
      "NOTICE: Finished analyzing 3000000 query variants\n",
      "NOTICE: Finished analyzing 4000000 query variants\n",
      "NOTICE: Reading FASTA sequences from /home/jupyter/annovar/humandb/hg38_refGeneMrna.fa ... Done with 1459 sequences\n",
      "WARNING: A total of 606 sequences will be ignored due to lack of correct ORF annotation\n",
      "NOTICE: Processing next batch with 5000000 unique variants in 5000000 input lines\n",
      "NOTICE: Finished analyzing 1000000 query variants\n",
      "NOTICE: Finished analyzing 2000000 query variants\n",
      "NOTICE: Finished analyzing 3000000 query variants\n",
      "NOTICE: Finished analyzing 4000000 query variants\n",
      "NOTICE: Reading FASTA sequences from /home/jupyter/annovar/humandb/hg38_refGeneMrna.fa ... Done with 1654 sequences\n",
      "WARNING: A total of 606 sequences will be ignored due to lack of correct ORF annotation\n",
      "NOTICE: Processing next batch with 5000000 unique variants in 5000000 input lines\n",
      "NOTICE: Finished analyzing 1000000 query variants\n",
      "NOTICE: Finished analyzing 2000000 query variants\n",
      "NOTICE: Finished analyzing 3000000 query variants\n",
      "NOTICE: Finished analyzing 4000000 query variants\n",
      "NOTICE: Reading FASTA sequences from /home/jupyter/annovar/humandb/hg38_refGeneMrna.fa ... Done with 992 sequences\n",
      "WARNING: A total of 606 sequences will be ignored due to lack of correct ORF annotation\n",
      "NOTICE: Processing next batch with 5000000 unique variants in 5000000 input lines\n",
      "NOTICE: Finished analyzing 1000000 query variants\n",
      "NOTICE: Finished analyzing 2000000 query variants\n",
      "NOTICE: Finished analyzing 3000000 query variants\n",
      "NOTICE: Finished analyzing 4000000 query variants\n",
      "NOTICE: Reading FASTA sequences from /home/jupyter/annovar/humandb/hg38_refGeneMrna.fa ... Done with 1162 sequences\n",
      "WARNING: A total of 606 sequences will be ignored due to lack of correct ORF annotation\n",
      "NOTICE: Processing next batch with 5000000 unique variants in 5000000 input lines\n",
      "NOTICE: Finished analyzing 1000000 query variants\n",
      "NOTICE: Finished analyzing 2000000 query variants\n",
      "NOTICE: Finished analyzing 3000000 query variants\n",
      "NOTICE: Finished analyzing 4000000 query variants\n",
      "NOTICE: Reading FASTA sequences from /home/jupyter/annovar/humandb/hg38_refGeneMrna.fa ... Done with 1292 sequences\n",
      "WARNING: A total of 606 sequences will be ignored due to lack of correct ORF annotation\n",
      "NOTICE: Processing next batch with 5000000 unique variants in 5000000 input lines\n",
      "NOTICE: Finished analyzing 1000000 query variants\n",
      "NOTICE: Finished analyzing 2000000 query variants\n",
      "NOTICE: Finished analyzing 3000000 query variants\n",
      "NOTICE: Finished analyzing 4000000 query variants\n",
      "NOTICE: Reading FASTA sequences from /home/jupyter/annovar/humandb/hg38_refGeneMrna.fa ... Done with 1070 sequences\n",
      "WARNING: A total of 606 sequences will be ignored due to lack of correct ORF annotation\n",
      "NOTICE: Processing next batch with 5000000 unique variants in 5000000 input lines\n",
      "NOTICE: Finished analyzing 1000000 query variants\n",
      "NOTICE: Finished analyzing 2000000 query variants\n",
      "NOTICE: Finished analyzing 3000000 query variants\n",
      "NOTICE: Finished analyzing 4000000 query variants\n",
      "NOTICE: Reading FASTA sequences from /home/jupyter/annovar/humandb/hg38_refGeneMrna.fa ... Done with 1258 sequences\n",
      "WARNING: A total of 606 sequences will be ignored due to lack of correct ORF annotation\n",
      "NOTICE: Processing next batch with 5000000 unique variants in 5000000 input lines\n",
      "NOTICE: Finished analyzing 1000000 query variants\n",
      "NOTICE: Finished analyzing 2000000 query variants\n",
      "NOTICE: Finished analyzing 3000000 query variants\n",
      "NOTICE: Finished analyzing 4000000 query variants\n",
      "NOTICE: Reading FASTA sequences from /home/jupyter/annovar/humandb/hg38_refGeneMrna.fa ... Done with 1029 sequences\n",
      "WARNING: A total of 606 sequences will be ignored due to lack of correct ORF annotation\n",
      "NOTICE: Processing next batch with 5000000 unique variants in 5000000 input lines\n",
      "NOTICE: Finished analyzing 1000000 query variants\n",
      "NOTICE: Finished analyzing 2000000 query variants\n",
      "NOTICE: Finished analyzing 3000000 query variants\n",
      "NOTICE: Finished analyzing 4000000 query variants\n",
      "NOTICE: Reading FASTA sequences from /home/jupyter/annovar/humandb/hg38_refGeneMrna.fa ... Done with 1318 sequences\n",
      "WARNING: A total of 606 sequences will be ignored due to lack of correct ORF annotation\n",
      "NOTICE: Processing next batch with 5000000 unique variants in 5000000 input lines\n",
      "NOTICE: Finished analyzing 1000000 query variants\n",
      "NOTICE: Finished analyzing 2000000 query variants\n",
      "NOTICE: Finished analyzing 3000000 query variants\n",
      "NOTICE: Finished analyzing 4000000 query variants\n",
      "NOTICE: Reading FASTA sequences from /home/jupyter/annovar/humandb/hg38_refGeneMrna.fa ... Done with 800 sequences\n",
      "WARNING: A total of 606 sequences will be ignored due to lack of correct ORF annotation\n",
      "NOTICE: Processing next batch with 5000000 unique variants in 5000000 input lines\n",
      "NOTICE: Finished analyzing 1000000 query variants\n",
      "NOTICE: Finished analyzing 2000000 query variants\n",
      "NOTICE: Finished analyzing 3000000 query variants\n",
      "NOTICE: Finished analyzing 4000000 query variants\n",
      "NOTICE: Reading FASTA sequences from /home/jupyter/annovar/humandb/hg38_refGeneMrna.fa ... Done with 627 sequences\n",
      "WARNING: A total of 606 sequences will be ignored due to lack of correct ORF annotation\n",
      "NOTICE: Processing next batch with 5000000 unique variants in 5000000 input lines\n",
      "NOTICE: Finished analyzing 1000000 query variants\n",
      "NOTICE: Finished analyzing 2000000 query variants\n",
      "NOTICE: Finished analyzing 3000000 query variants\n",
      "NOTICE: Finished analyzing 4000000 query variants\n",
      "NOTICE: Reading FASTA sequences from /home/jupyter/annovar/humandb/hg38_refGeneMrna.fa ... Done with 890 sequences\n",
      "WARNING: A total of 606 sequences will be ignored due to lack of correct ORF annotation\n",
      "NOTICE: Processing next batch with 5000000 unique variants in 5000000 input lines\n",
      "NOTICE: Finished analyzing 1000000 query variants\n",
      "NOTICE: Finished analyzing 2000000 query variants\n",
      "NOTICE: Finished analyzing 3000000 query variants\n",
      "NOTICE: Finished analyzing 4000000 query variants\n",
      "NOTICE: Reading FASTA sequences from /home/jupyter/annovar/humandb/hg38_refGeneMrna.fa ... Done with 896 sequences\n",
      "WARNING: A total of 606 sequences will be ignored due to lack of correct ORF annotation\n",
      "NOTICE: Processing next batch with 5000000 unique variants in 5000000 input lines\n",
      "NOTICE: Finished analyzing 1000000 query variants\n",
      "NOTICE: Finished analyzing 2000000 query variants\n",
      "NOTICE: Finished analyzing 3000000 query variants\n",
      "NOTICE: Finished analyzing 4000000 query variants\n",
      "NOTICE: Reading FASTA sequences from /home/jupyter/annovar/humandb/hg38_refGeneMrna.fa ... Done with 1547 sequences\n",
      "WARNING: A total of 606 sequences will be ignored due to lack of correct ORF annotation\n",
      "NOTICE: Processing next batch with 3893589 unique variants in 3893589 input lines\n",
      "NOTICE: Finished analyzing 1000000 query variants\n",
      "NOTICE: Finished analyzing 2000000 query variants\n",
      "NOTICE: Finished analyzing 3000000 query variants\n",
      "NOTICE: Reading FASTA sequences from /home/jupyter/annovar/humandb/hg38_refGeneMrna.fa ... Done with 606 sequences\n",
      "WARNING: A total of 606 sequences will be ignored due to lack of correct ORF annotation\n",
      "NOTICE: Variants with invalid input format are written to all_ancestries_annotated.refGene.invalid_input\n",
      "\n",
      "NOTICE: Running with system command <coding_change.pl  all_ancestries_annotated.refGene.exonic_variant_function.orig /home/jupyter/annovar/humandb//hg38_refGene.txt /home/jupyter/annovar/humandb//hg38_refGeneMrna.fa -alltranscript -out all_ancestries_annotated.refGene.fa -newevf all_ancestries_annotated.refGene.exonic_variant_function>\n",
      "WARNING: cannot find annotation for NM_024616 in the genefile /home/jupyter/annovar/humandb//hg38_refGene.txt or cannot infer the transcription start site\n",
      "WARNING: cannot find annotation for NM_024616 in the genefile /home/jupyter/annovar/humandb//hg38_refGene.txt or cannot infer the transcription start site\n",
      "WARNING: cannot find annotation for NM_024616 in the genefile /home/jupyter/annovar/humandb//hg38_refGene.txt or cannot infer the transcription start site\n",
      "Warning: 70 transcripts are flagged as having potential ORF issues (premature stopcodon or lack of stop codon)\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Processing operation=f protocol=avsnp151\n",
      "NOTICE: Finished reading 1 column headers for '-dbtype avsnp151'\n",
      "\n",
      "NOTICE: Running system command <annotate_variation.pl -filter -dbtype avsnp151 -buildver hg38 -outfile all_ancestries_annotated to_annotate.txt /home/jupyter/annovar/humandb/ -otherinfo>\n",
      "NOTICE: Output file with variants matching filtering criteria is written to all_ancestries_annotated.hg38_avsnp151_dropped, and output file with other variants is written to all_ancestries_annotated.hg38_avsnp151_filtered\n",
      "NOTICE: Processing next batch with 4733167 unique variants in 5000000 input lines\n",
      "NOTICE: Database index loaded. Total number of bins is 28835218 and the number of bins to be scanned is 574390\n",
      "NOTICE: Scanning filter database /home/jupyter/annovar/humandb/hg38_avsnp151.txt...Done\n",
      "NOTICE: Processing next batch with 4729047 unique variants in 5000000 input lines\n",
      "NOTICE: Database index loaded. Total number of bins is 28835218 and the number of bins to be scanned is 597682\n",
      "NOTICE: Scanning filter database /home/jupyter/annovar/humandb/hg38_avsnp151.txt...Done\n",
      "NOTICE: Processing next batch with 4730740 unique variants in 5000000 input lines\n",
      "NOTICE: Database index loaded. Total number of bins is 28835218 and the number of bins to be scanned is 587800\n",
      "NOTICE: Scanning filter database /home/jupyter/annovar/humandb/hg38_avsnp151.txt...Done\n",
      "NOTICE: Processing next batch with 4738742 unique variants in 5000000 input lines\n",
      "NOTICE: Database index loaded. Total number of bins is 28835218 and the number of bins to be scanned is 577879\n",
      "NOTICE: Scanning filter database /home/jupyter/annovar/humandb/hg38_avsnp151.txt...Done\n",
      "NOTICE: Processing next batch with 4741167 unique variants in 5000000 input lines\n",
      "NOTICE: Database index loaded. Total number of bins is 28835218 and the number of bins to be scanned is 563763\n",
      "NOTICE: Scanning filter database /home/jupyter/annovar/humandb/hg38_avsnp151.txt...Done\n",
      "NOTICE: Processing next batch with 4755884 unique variants in 5000000 input lines\n",
      "NOTICE: Database index loaded. Total number of bins is 28835218 and the number of bins to be scanned is 551741\n",
      "NOTICE: Scanning filter database /home/jupyter/annovar/humandb/hg38_avsnp151.txt...Done\n",
      "NOTICE: Processing next batch with 4774362 unique variants in 5000000 input lines\n",
      "NOTICE: Database index loaded. Total number of bins is 28835218 and the number of bins to be scanned is 478318\n",
      "NOTICE: Scanning filter database /home/jupyter/annovar/humandb/hg38_avsnp151.txt...Done\n",
      "NOTICE: Processing next batch with 4733871 unique variants in 5000000 input lines\n",
      "NOTICE: Database index loaded. Total number of bins is 28835218 and the number of bins to be scanned is 560645\n",
      "NOTICE: Scanning filter database /home/jupyter/annovar/humandb/hg38_avsnp151.txt...Done\n",
      "NOTICE: Processing next batch with 4733968 unique variants in 5000000 input lines\n",
      "NOTICE: Database index loaded. Total number of bins is 28835218 and the number of bins to be scanned is 531488\n",
      "NOTICE: Scanning filter database /home/jupyter/annovar/humandb/hg38_avsnp151.txt...Done\n",
      "NOTICE: Processing next batch with 4741508 unique variants in 5000000 input lines\n",
      "NOTICE: Database index loaded. Total number of bins is 28835218 and the number of bins to be scanned is 597376\n",
      "NOTICE: Scanning filter database /home/jupyter/annovar/humandb/hg38_avsnp151.txt...Done\n",
      "NOTICE: Processing next batch with 4732335 unique variants in 5000000 input lines\n",
      "NOTICE: Database index loaded. Total number of bins is 28835218 and the number of bins to be scanned is 584743\n",
      "NOTICE: Scanning filter database /home/jupyter/annovar/humandb/hg38_avsnp151.txt...Done\n",
      "NOTICE: Processing next batch with 4738262 unique variants in 5000000 input lines\n",
      "NOTICE: Database index loaded. Total number of bins is 28835218 and the number of bins to be scanned is 584831\n",
      "NOTICE: Scanning filter database /home/jupyter/annovar/humandb/hg38_avsnp151.txt...Done\n",
      "NOTICE: Processing next batch with 4733882 unique variants in 5000000 input lines\n",
      "NOTICE: Database index loaded. Total number of bins is 28835218 and the number of bins to be scanned is 590794\n",
      "NOTICE: Scanning filter database /home/jupyter/annovar/humandb/hg38_avsnp151.txt...Done\n",
      "NOTICE: Processing next batch with 4736378 unique variants in 5000000 input lines\n",
      "NOTICE: Database index loaded. Total number of bins is 28835218 and the number of bins to be scanned is 557212\n",
      "NOTICE: Scanning filter database /home/jupyter/annovar/humandb/hg38_avsnp151.txt...Done\n",
      "NOTICE: Processing next batch with 4737820 unique variants in 5000000 input lines\n",
      "NOTICE: Database index loaded. Total number of bins is 28835218 and the number of bins to be scanned is 539802\n",
      "NOTICE: Scanning filter database /home/jupyter/annovar/humandb/hg38_avsnp151.txt...Done\n",
      "NOTICE: Processing next batch with 4735123 unique variants in 5000000 input lines\n",
      "NOTICE: Database index loaded. Total number of bins is 28835218 and the number of bins to be scanned is 594403\n",
      "NOTICE: Scanning filter database /home/jupyter/annovar/humandb/hg38_avsnp151.txt...Done\n",
      "NOTICE: Processing next batch with 4721423 unique variants in 5000000 input lines\n",
      "NOTICE: Database index loaded. Total number of bins is 28835218 and the number of bins to be scanned is 610477\n",
      "NOTICE: Scanning filter database /home/jupyter/annovar/humandb/hg38_avsnp151.txt...Done\n",
      "NOTICE: Processing next batch with 4751792 unique variants in 5000000 input lines\n",
      "NOTICE: Database index loaded. Total number of bins is 28835218 and the number of bins to be scanned is 542881\n",
      "NOTICE: Scanning filter database /home/jupyter/annovar/humandb/hg38_avsnp151.txt...Done\n",
      "NOTICE: Processing next batch with 4744366 unique variants in 5000000 input lines\n",
      "NOTICE: Database index loaded. Total number of bins is 28835218 and the number of bins to be scanned is 575327\n",
      "NOTICE: Scanning filter database /home/jupyter/annovar/humandb/hg38_avsnp151.txt...Done\n",
      "NOTICE: Processing next batch with 4736859 unique variants in 5000000 input lines\n",
      "NOTICE: Database index loaded. Total number of bins is 28835218 and the number of bins to be scanned is 596652\n",
      "NOTICE: Scanning filter database /home/jupyter/annovar/humandb/hg38_avsnp151.txt...Done\n",
      "NOTICE: Processing next batch with 4733028 unique variants in 5000000 input lines\n",
      "NOTICE: Database index loaded. Total number of bins is 28835218 and the number of bins to be scanned is 575941\n",
      "NOTICE: Scanning filter database /home/jupyter/annovar/humandb/hg38_avsnp151.txt...Done\n",
      "NOTICE: Processing next batch with 4752291 unique variants in 5000000 input lines\n",
      "NOTICE: Database index loaded. Total number of bins is 28835218 and the number of bins to be scanned is 580360\n",
      "NOTICE: Scanning filter database /home/jupyter/annovar/humandb/hg38_avsnp151.txt...Done\n",
      "NOTICE: Processing next batch with 4731300 unique variants in 5000000 input lines\n",
      "NOTICE: Database index loaded. Total number of bins is 28835218 and the number of bins to be scanned is 591098\n",
      "NOTICE: Scanning filter database /home/jupyter/annovar/humandb/hg38_avsnp151.txt...Done\n",
      "NOTICE: Processing next batch with 4733231 unique variants in 5000000 input lines\n",
      "NOTICE: Database index loaded. Total number of bins is 28835218 and the number of bins to be scanned is 577611\n",
      "NOTICE: Scanning filter database /home/jupyter/annovar/humandb/hg38_avsnp151.txt...Done\n",
      "NOTICE: Processing next batch with 4745364 unique variants in 5000000 input lines\n",
      "NOTICE: Database index loaded. Total number of bins is 28835218 and the number of bins to be scanned is 555828\n",
      "NOTICE: Scanning filter database /home/jupyter/annovar/humandb/hg38_avsnp151.txt...Done\n",
      "NOTICE: Processing next batch with 4726801 unique variants in 5000000 input lines\n",
      "NOTICE: Database index loaded. Total number of bins is 28835218 and the number of bins to be scanned is 597782\n",
      "NOTICE: Scanning filter database /home/jupyter/annovar/humandb/hg38_avsnp151.txt...Done\n",
      "NOTICE: Processing next batch with 4735569 unique variants in 5000000 input lines\n",
      "NOTICE: Database index loaded. Total number of bins is 28835218 and the number of bins to be scanned is 588090\n",
      "NOTICE: Scanning filter database /home/jupyter/annovar/humandb/hg38_avsnp151.txt...Done\n",
      "NOTICE: Processing next batch with 4738957 unique variants in 5000000 input lines\n",
      "NOTICE: Database index loaded. Total number of bins is 28835218 and the number of bins to be scanned is 590559\n",
      "NOTICE: Scanning filter database /home/jupyter/annovar/humandb/hg38_avsnp151.txt...Done\n",
      "NOTICE: Processing next batch with 1216352 unique variants in 1285793 input lines\n",
      "NOTICE: Database index loaded. Total number of bins is 28835218 and the number of bins to be scanned is 150663\n",
      "NOTICE: Scanning filter database /home/jupyter/annovar/humandb/hg38_avsnp151.txt...Done\n",
      "NOTICE: Variants with invalid input format are written to all_ancestries_annotated.invalid_input\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Processing operation=f protocol=clinvar_20240917\n",
      "NOTICE: Finished reading 13 column headers for '-dbtype clinvar_20240917'\n",
      "\n",
      "NOTICE: Running system command <annotate_variation.pl -filter -dbtype clinvar_20240917 -buildver hg38 -outfile all_ancestries_annotated to_annotate.txt /home/jupyter/annovar/humandb/ -otherinfo>\n",
      "NOTICE: the --dbtype clinvar_20240917 is assumed to be in generic ANNOVAR database format\n",
      "NOTICE: Output file with variants matching filtering criteria is written to all_ancestries_annotated.hg38_clinvar_20240917_dropped, and output file with other variants is written to all_ancestries_annotated.hg38_clinvar_20240917_filtered\n",
      "NOTICE: Processing next batch with 4733167 unique variants in 5000000 input lines\n",
      "NOTICE: Database index loaded. Total number of bins is 174536 and the number of bins to be scanned is 4569\n",
      "NOTICE: Scanning filter database /home/jupyter/annovar/humandb/hg38_clinvar_20240917.txt...Done\n",
      "NOTICE: Processing next batch with 4729047 unique variants in 5000000 input lines\n",
      "NOTICE: Database index loaded. Total number of bins is 174536 and the number of bins to be scanned is 4318\n",
      "NOTICE: Scanning filter database /home/jupyter/annovar/humandb/hg38_clinvar_20240917.txt...Done\n",
      "NOTICE: Processing next batch with 4730740 unique variants in 5000000 input lines\n",
      "NOTICE: Database index loaded. Total number of bins is 174536 and the number of bins to be scanned is 3267\n",
      "NOTICE: Scanning filter database /home/jupyter/annovar/humandb/hg38_clinvar_20240917.txt...Done\n",
      "NOTICE: Processing next batch with 4738742 unique variants in 5000000 input lines\n",
      "NOTICE: Database index loaded. Total number of bins is 174536 and the number of bins to be scanned is 3775\n",
      "NOTICE: Scanning filter database /home/jupyter/annovar/humandb/hg38_clinvar_20240917.txt...Done\n",
      "NOTICE: Processing next batch with 4741167 unique variants in 5000000 input lines\n",
      "NOTICE: Database index loaded. Total number of bins is 174536 and the number of bins to be scanned is 4996\n",
      "NOTICE: Scanning filter database /home/jupyter/annovar/humandb/hg38_clinvar_20240917.txt...Done\n",
      "NOTICE: Processing next batch with 4755884 unique variants in 5000000 input lines\n",
      "NOTICE: Database index loaded. Total number of bins is 174536 and the number of bins to be scanned is 5266\n",
      "NOTICE: Scanning filter database /home/jupyter/annovar/humandb/hg38_clinvar_20240917.txt...Done\n",
      "NOTICE: Processing next batch with 4774362 unique variants in 5000000 input lines\n",
      "NOTICE: Database index loaded. Total number of bins is 174536 and the number of bins to be scanned is 4464\n",
      "NOTICE: Scanning filter database /home/jupyter/annovar/humandb/hg38_clinvar_20240917.txt...Done\n",
      "NOTICE: Processing next batch with 4733871 unique variants in 5000000 input lines\n",
      "NOTICE: Database index loaded. Total number of bins is 174536 and the number of bins to be scanned is 7843\n",
      "NOTICE: Scanning filter database /home/jupyter/annovar/humandb/hg38_clinvar_20240917.txt...Done\n",
      "NOTICE: Processing next batch with 4733968 unique variants in 5000000 input lines\n",
      "NOTICE: Database index loaded. Total number of bins is 174536 and the number of bins to be scanned is 8034\n",
      "NOTICE: Scanning filter database /home/jupyter/annovar/humandb/hg38_clinvar_20240917.txt...Done\n",
      "NOTICE: Processing next batch with 4741508 unique variants in 5000000 input lines\n",
      "NOTICE: Database index loaded. Total number of bins is 174536 and the number of bins to be scanned is 5801\n",
      "NOTICE: Scanning filter database /home/jupyter/annovar/humandb/hg38_clinvar_20240917.txt...Done\n",
      "NOTICE: Processing next batch with 4732335 unique variants in 5000000 input lines\n",
      "NOTICE: Database index loaded. Total number of bins is 174536 and the number of bins to be scanned is 4281\n",
      "NOTICE: Scanning filter database /home/jupyter/annovar/humandb/hg38_clinvar_20240917.txt...Done\n",
      "NOTICE: Processing next batch with 4738262 unique variants in 5000000 input lines\n",
      "NOTICE: Database index loaded. Total number of bins is 174536 and the number of bins to be scanned is 4879\n",
      "NOTICE: Scanning filter database /home/jupyter/annovar/humandb/hg38_clinvar_20240917.txt...Done\n",
      "NOTICE: Processing next batch with 4733882 unique variants in 5000000 input lines\n",
      "NOTICE: Database index loaded. Total number of bins is 174536 and the number of bins to be scanned is 2911\n",
      "NOTICE: Scanning filter database /home/jupyter/annovar/humandb/hg38_clinvar_20240917.txt...Done\n",
      "NOTICE: Processing next batch with 4736378 unique variants in 5000000 input lines\n",
      "NOTICE: Database index loaded. Total number of bins is 174536 and the number of bins to be scanned is 3365\n",
      "NOTICE: Scanning filter database /home/jupyter/annovar/humandb/hg38_clinvar_20240917.txt...Done\n",
      "NOTICE: Processing next batch with 4737820 unique variants in 5000000 input lines\n",
      "NOTICE: Database index loaded. Total number of bins is 174536 and the number of bins to be scanned is 5418\n",
      "NOTICE: Scanning filter database /home/jupyter/annovar/humandb/hg38_clinvar_20240917.txt...Done\n",
      "NOTICE: Processing next batch with 4735123 unique variants in 5000000 input lines\n",
      "NOTICE: Database index loaded. Total number of bins is 174536 and the number of bins to be scanned is 2457\n",
      "NOTICE: Scanning filter database /home/jupyter/annovar/humandb/hg38_clinvar_20240917.txt...Done\n",
      "NOTICE: Processing next batch with 4721423 unique variants in 5000000 input lines\n",
      "NOTICE: Database index loaded. Total number of bins is 174536 and the number of bins to be scanned is 3854\n",
      "NOTICE: Scanning filter database /home/jupyter/annovar/humandb/hg38_clinvar_20240917.txt...Done\n",
      "NOTICE: Processing next batch with 4751792 unique variants in 5000000 input lines\n",
      "NOTICE: Database index loaded. Total number of bins is 174536 and the number of bins to be scanned is 4175\n",
      "NOTICE: Scanning filter database /home/jupyter/annovar/humandb/hg38_clinvar_20240917.txt...Done\n",
      "NOTICE: Processing next batch with 4744366 unique variants in 5000000 input lines\n",
      "NOTICE: Database index loaded. Total number of bins is 174536 and the number of bins to be scanned is 3197\n",
      "NOTICE: Scanning filter database /home/jupyter/annovar/humandb/hg38_clinvar_20240917.txt...Done\n",
      "NOTICE: Processing next batch with 4736859 unique variants in 5000000 input lines\n",
      "NOTICE: Database index loaded. Total number of bins is 174536 and the number of bins to be scanned is 3684\n",
      "NOTICE: Scanning filter database /home/jupyter/annovar/humandb/hg38_clinvar_20240917.txt...Done\n",
      "NOTICE: Processing next batch with 4733028 unique variants in 5000000 input lines\n",
      "NOTICE: Database index loaded. Total number of bins is 174536 and the number of bins to be scanned is 2446\n",
      "NOTICE: Scanning filter database /home/jupyter/annovar/humandb/hg38_clinvar_20240917.txt...Done\n",
      "NOTICE: Processing next batch with 4752291 unique variants in 5000000 input lines\n",
      "NOTICE: Database index loaded. Total number of bins is 174536 and the number of bins to be scanned is 3686\n",
      "NOTICE: Scanning filter database /home/jupyter/annovar/humandb/hg38_clinvar_20240917.txt...Done\n",
      "NOTICE: Processing next batch with 4731300 unique variants in 5000000 input lines\n",
      "NOTICE: Database index loaded. Total number of bins is 174536 and the number of bins to be scanned is 2033\n",
      "NOTICE: Scanning filter database /home/jupyter/annovar/humandb/hg38_clinvar_20240917.txt...Done\n",
      "NOTICE: Processing next batch with 4733231 unique variants in 5000000 input lines\n",
      "NOTICE: Database index loaded. Total number of bins is 174536 and the number of bins to be scanned is 2063\n",
      "NOTICE: Scanning filter database /home/jupyter/annovar/humandb/hg38_clinvar_20240917.txt...Done\n",
      "NOTICE: Processing next batch with 4745364 unique variants in 5000000 input lines\n",
      "NOTICE: Database index loaded. Total number of bins is 174536 and the number of bins to be scanned is 2382\n",
      "NOTICE: Scanning filter database /home/jupyter/annovar/humandb/hg38_clinvar_20240917.txt...Done\n",
      "NOTICE: Processing next batch with 4726801 unique variants in 5000000 input lines\n",
      "NOTICE: Database index loaded. Total number of bins is 174536 and the number of bins to be scanned is 2844\n",
      "NOTICE: Scanning filter database /home/jupyter/annovar/humandb/hg38_clinvar_20240917.txt...Done\n",
      "NOTICE: Processing next batch with 4735569 unique variants in 5000000 input lines\n",
      "NOTICE: Database index loaded. Total number of bins is 174536 and the number of bins to be scanned is 3038\n",
      "NOTICE: Scanning filter database /home/jupyter/annovar/humandb/hg38_clinvar_20240917.txt...Done\n",
      "NOTICE: Processing next batch with 4738957 unique variants in 5000000 input lines\n",
      "NOTICE: Database index loaded. Total number of bins is 174536 and the number of bins to be scanned is 3933\n",
      "NOTICE: Scanning filter database /home/jupyter/annovar/humandb/hg38_clinvar_20240917.txt...Done\n",
      "NOTICE: Processing next batch with 1216352 unique variants in 1285793 input lines\n",
      "NOTICE: Database index loaded. Total number of bins is 174536 and the number of bins to be scanned is 459\n",
      "NOTICE: Scanning filter database /home/jupyter/annovar/humandb/hg38_clinvar_20240917.txt...Done\n",
      "NOTICE: Variants with invalid input format are written to all_ancestries_annotated.invalid_input\n",
      "-----------------------------------------------------------------\n",
      "NOTICE: Multianno output file is written to all_ancestries_annotated.hg38_multianno.txt\n"
     ]
    }
   ],
   "source": [
    "# command to run with just clinvar annotations (no CAD scores)\n",
    "\n",
    "!/home/jupyter/annovar/table_annovar.pl to_annotate.txt /home/jupyter/annovar/humandb/ -buildver hg38 -out all_ancestries_annotated -protocol refGene,avsnp151,clinvar_20240917 -operation g,f,f -nastring ."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "977c273f-9524-451e-a154-4d14fb9d5fb0",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "tags": []
   },
   "outputs": [],
   "source": [
    "# command to run CADD score annotations\n",
    "# - note: this was run after the clinvar annotations were run and manually appended to the ClinVar annotations\n",
    "#         to run this in the initial run with the ClinVar annotations, include the database in the prior command\n",
    "\n",
    "# !/home/jupyter/annovar/table_annovar.pl to_annotate.txt /home/jupyter/annovar/humandb/ -buildver hg38 -out all_ancestries_annotated_cadd -protocol refGene,avsnp151,dbnsfp33a -operation g,f,f -nastring ."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "78dc3d16-c33d-4a73-965a-fca523a20ef2",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-rw-r--r-- 1 jupyter jupyter 2.5G Oct  9 23:11 all_ancestries.txt\n",
      "-rw-r--r-- 1 jupyter jupyter 4.1G Oct 10 17:06 all_ancestries_annotated.hg38_avsnp150_dropped\n",
      "-rw-r--r-- 1 jupyter jupyter 1.2G Oct 10 17:06 all_ancestries_annotated.hg38_avsnp150_filtered\n",
      "-rw-r--r-- 1 jupyter jupyter 5.6G Oct 16 06:54 all_ancestries_annotated.hg38_avsnp151_dropped\n",
      "-rw-r--r-- 1 jupyter jupyter 348M Oct 16 06:54 all_ancestries_annotated.hg38_avsnp151_filtered\n",
      "-rw-r--r-- 1 jupyter jupyter  85M Oct 10 17:49 all_ancestries_annotated.hg38_clinvar_20221231_dropped\n",
      "-rw-r--r-- 1 jupyter jupyter 3.5G Oct 10 17:49 all_ancestries_annotated.hg38_clinvar_20221231_filtered\n",
      "-rw-r--r-- 1 jupyter jupyter 138M Oct 16 07:40 all_ancestries_annotated.hg38_clinvar_20240917_dropped\n",
      "-rw-r--r-- 1 jupyter jupyter 3.5G Oct 16 07:40 all_ancestries_annotated.hg38_clinvar_20240917_filtered\n",
      "-rw-r--r-- 1 jupyter jupyter 412M Oct 11 01:13 all_ancestries_annotated.hg38_dbnsfp33a_dropped\n",
      "-rw-r--r-- 1 jupyter jupyter 3.5G Oct 11 01:13 all_ancestries_annotated.hg38_dbnsfp33a_filtered\n",
      "-rw-r--r-- 1 jupyter jupyter  14G Oct 16 08:22 all_ancestries_annotated.hg38_multianno.txt\n",
      "-rw-r--r-- 1 jupyter jupyter 221M Oct 16 07:40 all_ancestries_annotated.invalid_input\n",
      "-rw-r--r-- 1 jupyter jupyter 8.9K Oct 16 07:40 all_ancestries_annotated.log\n",
      "-rw-r--r-- 1 jupyter jupyter 279M Oct 16 05:08 all_ancestries_annotated.refGene.exonic_variant_function\n",
      "-rw-r--r-- 1 jupyter jupyter 277M Oct 16 05:06 all_ancestries_annotated.refGene.exonic_variant_function.orig\n",
      "-rw-r--r-- 1 jupyter jupyter  69M Oct 16 05:08 all_ancestries_annotated.refGene.fa\n",
      "-rw-r--r-- 1 jupyter jupyter 221M Oct 16 05:06 all_ancestries_annotated.refGene.invalid_input\n",
      "-rw-r--r-- 1 jupyter jupyter  14K Oct 16 05:06 all_ancestries_annotated.refGene.log\n",
      "-rw-r--r-- 1 jupyter jupyter 7.9G Oct 16 05:06 all_ancestries_annotated.refGene.variant_function\n"
     ]
    }
   ],
   "source": [
    "# check that annotation files were created\n",
    "\n",
    "!ls -lh all_ancestries*"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "df3b960d-6ae9-4dfe-8d1c-f9b2158734ce",
   "metadata": {
    "tags": []
   },
   "source": [
    "## filter for pathogenic variants"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "d672b610-9dd6-4bfe-8283-f0a737fec4ca",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# genes of interest (PD genes)\n",
    "genes = ['SNCA', 'PRKN', 'PARK7', 'LRRK2', 'PINK1', 'ATP13A2', 'FBX07', 'GBA', 'PLA2G6', 'VPS35', 'POLG', 'DNAJC6', 'SYNJ1', 'VPS13C', \\\n",
    "         'UCHL1', 'HTRA2', 'GIGYF2', 'EIF4G1', 'DNAJC13', 'TMEM230', 'LRP10']\n",
    "\n",
    "# create file with these genes to awk from\n",
    "with open(f'{workspace}/genes_of_interest.txt', 'w') as f:\n",
    "    for gene in genes:\n",
    "        f.write(f'{gene}\\n')\n",
    "f.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "95633244-b9fc-4dcf-8e28-ab8532d27c7a",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# get only variants that are within genes of interest\n",
    "\n",
    "!grep -F -f {workspace}/genes_of_interest.txt {workspace}/bfiles/all_ancestries_annotated.hg38_multianno.txt > {workspace}/annotated_selectedgenes.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "691fb2c5-e12d-4d92-b3a9-c0bd4bbc807e",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['Chr', 'Start', 'End', 'Ref', 'Alt', 'Func.refGene', 'Gene.refGene', 'GeneDetail.refGene', 'ExonicFunc.refGene', 'AAChange.refGene', 'avsnp151', 'CLNALLELEID', 'CLNDN', 'CLNDISDB', 'CLNREVSTAT', 'CLNSIG', 'ONCDN', 'ONCDISDB', 'ONCREVSTAT', 'ONC', 'SCIDN', 'SCIDISDB', 'SCIREVSTAT', 'SCI\\n']\n"
     ]
    }
   ],
   "source": [
    "# get column names\n",
    "annotated_selected_genes_file = open(f'{workspace}/bfiles/all_ancestries_annotated.hg38_multianno.txt', 'r')\n",
    "col_names = annotated_selected_genes_file.readline().split('\\t')\n",
    "print(col_names)\n",
    "\n",
    "want_cols = ['Chr','Start','End','Ref','Alt','Gene.refGene','avsnp151','CLNALLELEID','CLNDN','CLNDISDB','CLNREVSTAT','CLNSIG']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "id": "dcef0ae2-bd27-4cfd-8f82-3841657b5fbd",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>Gene.refGene</th>\n",
       "      <th>avsnp151</th>\n",
       "      <th>CLNALLELEID</th>\n",
       "      <th>CLNDN</th>\n",
       "      <th>CLNDISDB</th>\n",
       "      <th>CLNREVSTAT</th>\n",
       "      <th>CLNSIG</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr12</td>\n",
       "      <td>40168848</td>\n",
       "      <td>40168848</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>LINC02471;LRRK2</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr12</td>\n",
       "      <td>40168854</td>\n",
       "      <td>40168854</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>LINC02471;LRRK2</td>\n",
       "      <td>rs541787948</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr12</td>\n",
       "      <td>40168868</td>\n",
       "      <td>40168868</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>LINC02471;LRRK2</td>\n",
       "      <td>rs535940699</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr12</td>\n",
       "      <td>40168875</td>\n",
       "      <td>40168875</td>\n",
       "      <td>T</td>\n",
       "      <td>TC</td>\n",
       "      <td>LINC02471;LRRK2</td>\n",
       "      <td>rs576168793</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr12</td>\n",
       "      <td>40168880</td>\n",
       "      <td>40168880</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>LINC02471;LRRK2</td>\n",
       "      <td>rs1005738941</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>464769</th>\n",
       "      <td>chr6</td>\n",
       "      <td>162727742</td>\n",
       "      <td>162727742</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>PRKN</td>\n",
       "      <td>rs1362063712</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>464770</th>\n",
       "      <td>chr6</td>\n",
       "      <td>162727743</td>\n",
       "      <td>162727743</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>PRKN</td>\n",
       "      <td>rs751054800</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>464771</th>\n",
       "      <td>chr6</td>\n",
       "      <td>162727745</td>\n",
       "      <td>162727745</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>PRKN</td>\n",
       "      <td>rs756731328</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>464772</th>\n",
       "      <td>chr6</td>\n",
       "      <td>162727753</td>\n",
       "      <td>162727753</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>PRKN</td>\n",
       "      <td>rs556535340</td>\n",
       "      <td>895664</td>\n",
       "      <td>Autosomal_recessive_juvenile_Parkinson_disease_2</td>\n",
       "      <td>MONDO:MONDO:0010820,MedGen:C1868675,OMIM:60011...</td>\n",
       "      <td>criteria_provided,_single_submitter</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>464773</th>\n",
       "      <td>chr6</td>\n",
       "      <td>162727764</td>\n",
       "      <td>162727764</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>PRKN</td>\n",
       "      <td>rs1232075505</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>464774 rows Ã— 12 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          Chr      Start        End Ref Alt     Gene.refGene      avsnp151  \\\n",
       "0       chr12   40168848   40168848   A   G  LINC02471;LRRK2             .   \n",
       "1       chr12   40168854   40168854   C   T  LINC02471;LRRK2   rs541787948   \n",
       "2       chr12   40168868   40168868   C   A  LINC02471;LRRK2   rs535940699   \n",
       "3       chr12   40168875   40168875   T  TC  LINC02471;LRRK2   rs576168793   \n",
       "4       chr12   40168880   40168880   C   A  LINC02471;LRRK2  rs1005738941   \n",
       "...       ...        ...        ...  ..  ..              ...           ...   \n",
       "464769   chr6  162727742  162727742   C   G             PRKN  rs1362063712   \n",
       "464770   chr6  162727743  162727743   A   G             PRKN   rs751054800   \n",
       "464771   chr6  162727745  162727745   C   A             PRKN   rs756731328   \n",
       "464772   chr6  162727753  162727753   C   T             PRKN   rs556535340   \n",
       "464773   chr6  162727764  162727764   C   T             PRKN  rs1232075505   \n",
       "\n",
       "       CLNALLELEID                                             CLNDN  \\\n",
       "0                .                                                 .   \n",
       "1                .                                                 .   \n",
       "2                .                                                 .   \n",
       "3                .                                                 .   \n",
       "4                .                                                 .   \n",
       "...            ...                                               ...   \n",
       "464769           .                                                 .   \n",
       "464770           .                                                 .   \n",
       "464771           .                                                 .   \n",
       "464772      895664  Autosomal_recessive_juvenile_Parkinson_disease_2   \n",
       "464773           .                                                 .   \n",
       "\n",
       "                                                 CLNDISDB  \\\n",
       "0                                                       .   \n",
       "1                                                       .   \n",
       "2                                                       .   \n",
       "3                                                       .   \n",
       "4                                                       .   \n",
       "...                                                   ...   \n",
       "464769                                                  .   \n",
       "464770                                                  .   \n",
       "464771                                                  .   \n",
       "464772  MONDO:MONDO:0010820,MedGen:C1868675,OMIM:60011...   \n",
       "464773                                                  .   \n",
       "\n",
       "                                 CLNREVSTAT                  CLNSIG  \n",
       "0                                         .                       .  \n",
       "1                                         .                       .  \n",
       "2                                         .                       .  \n",
       "3                                         .                       .  \n",
       "4                                         .                       .  \n",
       "...                                     ...                     ...  \n",
       "464769                                    .                       .  \n",
       "464770                                    .                       .  \n",
       "464771                                    .                       .  \n",
       "464772  criteria_provided,_single_submitter  Uncertain_significance  \n",
       "464773                                    .                       .  \n",
       "\n",
       "[464774 rows x 12 columns]"
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "annotated_select_genes = pd.read_csv(f'{workspace}/annotated_selectedgenes.txt', sep='\\t', header=None, \\\n",
    "                                     names=col_names, usecols=want_cols)\n",
    "annotated_select_genes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "90ad7c48-35b8-4f5b-8d81-a9aff7e83e54",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(34, 24)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>Func.refGene</th>\n",
       "      <th>Gene.refGene</th>\n",
       "      <th>GeneDetail.refGene</th>\n",
       "      <th>ExonicFunc.refGene</th>\n",
       "      <th>AAChange.refGene</th>\n",
       "      <th>...</th>\n",
       "      <th>CLNREVSTAT</th>\n",
       "      <th>CLNSIG</th>\n",
       "      <th>ONCDN</th>\n",
       "      <th>ONCDISDB</th>\n",
       "      <th>ONCREVSTAT</th>\n",
       "      <th>ONC</th>\n",
       "      <th>SCIDN</th>\n",
       "      <th>SCIDISDB</th>\n",
       "      <th>SCIREVSTAT</th>\n",
       "      <th>SCI\\n</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>9898</th>\n",
       "      <td>chr12</td>\n",
       "      <td>40310434</td>\n",
       "      <td>40310434</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>exonic</td>\n",
       "      <td>LRRK2</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>LRRK2:NM_198578:exon31:c.C4321T:p.R1441C</td>\n",
       "      <td>...</td>\n",
       "      <td>criteria_provided,_multiple_submitters,_no_con...</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47879</th>\n",
       "      <td>chr15</td>\n",
       "      <td>89319225</td>\n",
       "      <td>89319225</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>intronic</td>\n",
       "      <td>POLG</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>criteria_provided,_multiple_submitters,_no_con...</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48145</th>\n",
       "      <td>chr15</td>\n",
       "      <td>89321780</td>\n",
       "      <td>89321780</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>exonic</td>\n",
       "      <td>POLG</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>POLG:NM_001126131:exon16:c.C2554T:p.R852C,POLG...</td>\n",
       "      <td>...</td>\n",
       "      <td>criteria_provided,_multiple_submitters,_no_con...</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48146</th>\n",
       "      <td>chr15</td>\n",
       "      <td>89321792</td>\n",
       "      <td>89321792</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>exonic</td>\n",
       "      <td>POLG</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>POLG:NM_001126131:exon16:c.G2542A:p.G848S,POLG...</td>\n",
       "      <td>...</td>\n",
       "      <td>criteria_provided,_multiple_submitters,_no_con...</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48530</th>\n",
       "      <td>chr15</td>\n",
       "      <td>89325610</td>\n",
       "      <td>89325610</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>exonic</td>\n",
       "      <td>POLG</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>POLG:NM_001126131:exon10:c.C1789T:p.R597W,POLG...</td>\n",
       "      <td>...</td>\n",
       "      <td>criteria_provided,_multiple_submitters,_no_con...</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48542</th>\n",
       "      <td>chr15</td>\n",
       "      <td>89325679</td>\n",
       "      <td>89325679</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>exonic</td>\n",
       "      <td>POLG</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>POLG:NM_001126131:exon10:c.C1720T:p.R574W,POLG...</td>\n",
       "      <td>...</td>\n",
       "      <td>criteria_provided,_multiple_submitters,_no_con...</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48682</th>\n",
       "      <td>chr15</td>\n",
       "      <td>89327166</td>\n",
       "      <td>89327166</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>splicing</td>\n",
       "      <td>POLG</td>\n",
       "      <td>NM_001126131:exon7:c.1433+1G&gt;A;NM_002693:exon7...</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>criteria_provided,_multiple_submitters,_no_con...</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48685</th>\n",
       "      <td>chr15</td>\n",
       "      <td>89327201</td>\n",
       "      <td>89327201</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>exonic</td>\n",
       "      <td>POLG</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>POLG:NM_001126131:exon7:c.G1399A:p.A467T,POLG:...</td>\n",
       "      <td>...</td>\n",
       "      <td>criteria_provided,_multiple_submitters,_no_con...</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48843</th>\n",
       "      <td>chr15</td>\n",
       "      <td>89329041</td>\n",
       "      <td>89329041</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>exonic</td>\n",
       "      <td>POLG</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>POLG:NM_001126131:exon4:c.C925T:p.R309C,POLG:N...</td>\n",
       "      <td>...</td>\n",
       "      <td>criteria_provided,_multiple_submitters,_no_con...</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70751</th>\n",
       "      <td>chr1</td>\n",
       "      <td>155235196</td>\n",
       "      <td>155235196</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>exonic</td>\n",
       "      <td>GBA</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>GBA:NM_001171811:exon9:c.C1243T:p.R415C,GBA:NM...</td>\n",
       "      <td>...</td>\n",
       "      <td>criteria_provided,_multiple_submitters,_no_con...</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70835</th>\n",
       "      <td>chr1</td>\n",
       "      <td>155236295</td>\n",
       "      <td>155236295</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>exonic</td>\n",
       "      <td>GBA</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>GBA:NM_001171811:exon7:c.C913T:p.R305W,GBA:NM_...</td>\n",
       "      <td>...</td>\n",
       "      <td>criteria_provided,_single_submitter</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70948</th>\n",
       "      <td>chr1</td>\n",
       "      <td>155237453</td>\n",
       "      <td>155237453</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>exonic</td>\n",
       "      <td>GBA</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>GBA:NM_001171811:exon6:c.G626A:p.R209Q,GBA:NM_...</td>\n",
       "      <td>...</td>\n",
       "      <td>criteria_provided,_multiple_submitters,_no_con...</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>71013</th>\n",
       "      <td>chr1</td>\n",
       "      <td>155238206</td>\n",
       "      <td>155238206</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>exonic</td>\n",
       "      <td>GBA</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>GBA:NM_001171811:exon5:c.T428G:p.V143G,GBA:NM_...</td>\n",
       "      <td>...</td>\n",
       "      <td>criteria_provided,_single_submitter</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>71017</th>\n",
       "      <td>chr1</td>\n",
       "      <td>155238215</td>\n",
       "      <td>155238215</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>exonic</td>\n",
       "      <td>GBA</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>GBA:NM_001171811:exon5:c.A419G:p.N140S,GBA:NM_...</td>\n",
       "      <td>...</td>\n",
       "      <td>criteria_provided,_multiple_submitters,_no_con...</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>71038</th>\n",
       "      <td>chr1</td>\n",
       "      <td>155238630</td>\n",
       "      <td>155238630</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>exonic</td>\n",
       "      <td>GBA</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>GBA:NM_001171811:exon4:c.C214T:p.R72W,GBA:NM_0...</td>\n",
       "      <td>...</td>\n",
       "      <td>criteria_provided,_multiple_submitters,_no_con...</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>71134</th>\n",
       "      <td>chr1</td>\n",
       "      <td>155239934</td>\n",
       "      <td>155239934</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>exonic</td>\n",
       "      <td>GBA</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>GBA:NM_000157:exon3:c.C259T:p.R87W,GBA:NM_0011...</td>\n",
       "      <td>...</td>\n",
       "      <td>criteria_provided,_multiple_submitters,_no_con...</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>72210</th>\n",
       "      <td>chr1</td>\n",
       "      <td>16988455</td>\n",
       "      <td>16988455</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>exonic</td>\n",
       "      <td>ATP13A2</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>ATP13A2:NM_001141974:exon23:c.G2497A:p.G833R,A...</td>\n",
       "      <td>...</td>\n",
       "      <td>no_assertion_criteria_provided</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>72328</th>\n",
       "      <td>chr1</td>\n",
       "      <td>16989961</td>\n",
       "      <td>16989961</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>exonic</td>\n",
       "      <td>ATP13A2</td>\n",
       "      <td>.</td>\n",
       "      <td>stopgain</td>\n",
       "      <td>ATP13A2:NM_001141973:exon22:c.C2440T:p.R814X,A...</td>\n",
       "      <td>...</td>\n",
       "      <td>criteria_provided,_multiple_submitters,_no_con...</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>72536</th>\n",
       "      <td>chr1</td>\n",
       "      <td>16992345</td>\n",
       "      <td>16992345</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>exonic</td>\n",
       "      <td>ATP13A2</td>\n",
       "      <td>.</td>\n",
       "      <td>stopgain</td>\n",
       "      <td>ATP13A2:NM_001141973:exon18:c.C1888T:p.Q630X,A...</td>\n",
       "      <td>...</td>\n",
       "      <td>criteria_provided,_multiple_submitters,_no_con...</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>72904</th>\n",
       "      <td>chr1</td>\n",
       "      <td>16996059</td>\n",
       "      <td>16996059</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>exonic</td>\n",
       "      <td>ATP13A2</td>\n",
       "      <td>.</td>\n",
       "      <td>stopgain</td>\n",
       "      <td>ATP13A2:NM_001141973:exon15:c.C1444T:p.R482X,A...</td>\n",
       "      <td>...</td>\n",
       "      <td>criteria_provided,_multiple_submitters,_no_con...</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>73729</th>\n",
       "      <td>chr1</td>\n",
       "      <td>17005450</td>\n",
       "      <td>17005450</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>exonic</td>\n",
       "      <td>ATP13A2</td>\n",
       "      <td>.</td>\n",
       "      <td>stopgain</td>\n",
       "      <td>ATP13A2:NM_001141973:exon3:c.G212A:p.W71X,ATP1...</td>\n",
       "      <td>...</td>\n",
       "      <td>criteria_provided,_multiple_submitters,_no_con...</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>76909</th>\n",
       "      <td>chr1</td>\n",
       "      <td>20645640</td>\n",
       "      <td>20645640</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>exonic</td>\n",
       "      <td>PINK1</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>PINK1:NM_032409:exon5:c.T1040C:p.L347P</td>\n",
       "      <td>...</td>\n",
       "      <td>criteria_provided,_multiple_submitters,_no_con...</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>77218</th>\n",
       "      <td>chr1</td>\n",
       "      <td>20649109</td>\n",
       "      <td>20649109</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>exonic</td>\n",
       "      <td>PINK1</td>\n",
       "      <td>.</td>\n",
       "      <td>stopgain</td>\n",
       "      <td>PINK1:NM_032409:exon7:c.C1366T:p.Q456X</td>\n",
       "      <td>...</td>\n",
       "      <td>criteria_provided,_multiple_submitters,_no_con...</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>77228</th>\n",
       "      <td>chr1</td>\n",
       "      <td>20649217</td>\n",
       "      <td>20649217</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>exonic</td>\n",
       "      <td>PINK1</td>\n",
       "      <td>.</td>\n",
       "      <td>stopgain</td>\n",
       "      <td>PINK1:NM_032409:exon7:c.C1474T:p.R492X</td>\n",
       "      <td>...</td>\n",
       "      <td>criteria_provided,_multiple_submitters,_no_con...</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>104335</th>\n",
       "      <td>chr20</td>\n",
       "      <td>5100921</td>\n",
       "      <td>5100921</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>exonic</td>\n",
       "      <td>TMEM230</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>TMEM230:NM_001009925:exon4:c.G233T:p.R78L,TMEM...</td>\n",
       "      <td>...</td>\n",
       "      <td>criteria_provided,_single_submitter</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>113919</th>\n",
       "      <td>chr21</td>\n",
       "      <td>32726907</td>\n",
       "      <td>32726907</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>exonic</td>\n",
       "      <td>SYNJ1</td>\n",
       "      <td>.</td>\n",
       "      <td>stopgain</td>\n",
       "      <td>SYNJ1:NM_003895:exon2:c.G106T:p.E36X</td>\n",
       "      <td>...</td>\n",
       "      <td>criteria_provided,_single_submitter</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>114566</th>\n",
       "      <td>chr22</td>\n",
       "      <td>38115658</td>\n",
       "      <td>38115658</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>exonic</td>\n",
       "      <td>PLA2G6</td>\n",
       "      <td>.</td>\n",
       "      <td>stopgain</td>\n",
       "      <td>PLA2G6:NM_001004426:exon13:c.C1741T:p.R581X,PL...</td>\n",
       "      <td>...</td>\n",
       "      <td>criteria_provided,_multiple_submitters,_no_con...</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>116177</th>\n",
       "      <td>chr22</td>\n",
       "      <td>38132917</td>\n",
       "      <td>38132917</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>exonic</td>\n",
       "      <td>PLA2G6</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>PLA2G6:NM_001349868:exon6:c.G313T:p.D105Y,PLA2...</td>\n",
       "      <td>...</td>\n",
       "      <td>criteria_provided,_multiple_submitters,_no_con...</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>119356</th>\n",
       "      <td>chr22</td>\n",
       "      <td>38169318</td>\n",
       "      <td>38169318</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>exonic</td>\n",
       "      <td>PLA2G6</td>\n",
       "      <td>.</td>\n",
       "      <td>stopgain</td>\n",
       "      <td>PLA2G6:NM_001004426:exon2:c.C109T:p.R37X,PLA2G...</td>\n",
       "      <td>...</td>\n",
       "      <td>criteria_provided,_multiple_submitters,_no_con...</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>346356</th>\n",
       "      <td>chr6</td>\n",
       "      <td>161350214</td>\n",
       "      <td>161350214</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>intronic</td>\n",
       "      <td>PRKN</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>criteria_provided,_single_submitter</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>382697</th>\n",
       "      <td>chr6</td>\n",
       "      <td>161785793</td>\n",
       "      <td>161785793</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>exonic</td>\n",
       "      <td>PRKN</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>PRKN:NM_013988:exon4:c.G403C:p.G135R,PRKN:NM_0...</td>\n",
       "      <td>...</td>\n",
       "      <td>criteria_provided,_single_submitter</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>382708</th>\n",
       "      <td>chr6</td>\n",
       "      <td>161785839</td>\n",
       "      <td>161785839</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>exonic</td>\n",
       "      <td>PRKN</td>\n",
       "      <td>.</td>\n",
       "      <td>stopgain</td>\n",
       "      <td>PRKN:NM_013988:exon4:c.T357A:p.C119X,PRKN:NM_0...</td>\n",
       "      <td>...</td>\n",
       "      <td>criteria_provided,_multiple_submitters,_no_con...</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>440038</th>\n",
       "      <td>chr6</td>\n",
       "      <td>162443314</td>\n",
       "      <td>162443314</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>exonic</td>\n",
       "      <td>PRKN</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>PRKN:NM_004562:exon2:c.T167A:p.V56E,PRKN:NM_01...</td>\n",
       "      <td>...</td>\n",
       "      <td>criteria_provided,_multiple_submitters,_no_con...</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>440042</th>\n",
       "      <td>chr6</td>\n",
       "      <td>162443356</td>\n",
       "      <td>162443356</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>exonic</td>\n",
       "      <td>PRKN</td>\n",
       "      <td>.</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>PRKN:NM_004562:exon2:c.G125C:p.R42P,PRKN:NM_01...</td>\n",
       "      <td>...</td>\n",
       "      <td>criteria_provided,_single_submitter</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>34 rows Ã— 24 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          Chr      Start        End Ref Alt Func.refGene Gene.refGene  \\\n",
       "9898    chr12   40310434   40310434   C   T       exonic        LRRK2   \n",
       "47879   chr15   89319225   89319225   T   A     intronic         POLG   \n",
       "48145   chr15   89321780   89321780   G   A       exonic         POLG   \n",
       "48146   chr15   89321792   89321792   C   T       exonic         POLG   \n",
       "48530   chr15   89325610   89325610   G   A       exonic         POLG   \n",
       "48542   chr15   89325679   89325679   G   A       exonic         POLG   \n",
       "48682   chr15   89327166   89327166   C   T     splicing         POLG   \n",
       "48685   chr15   89327201   89327201   C   T       exonic         POLG   \n",
       "48843   chr15   89329041   89329041   G   A       exonic         POLG   \n",
       "70751    chr1  155235196  155235196   G   A       exonic          GBA   \n",
       "70835    chr1  155236295  155236295   G   A       exonic          GBA   \n",
       "70948    chr1  155237453  155237453   C   T       exonic          GBA   \n",
       "71013    chr1  155238206  155238206   A   C       exonic          GBA   \n",
       "71017    chr1  155238215  155238215   T   C       exonic          GBA   \n",
       "71038    chr1  155238630  155238630   G   A       exonic          GBA   \n",
       "71134    chr1  155239934  155239934   G   A       exonic          GBA   \n",
       "72210    chr1   16988455   16988455   C   T       exonic      ATP13A2   \n",
       "72328    chr1   16989961   16989961   G   A       exonic      ATP13A2   \n",
       "72536    chr1   16992345   16992345   G   A       exonic      ATP13A2   \n",
       "72904    chr1   16996059   16996059   G   A       exonic      ATP13A2   \n",
       "73729    chr1   17005450   17005450   C   T       exonic      ATP13A2   \n",
       "76909    chr1   20645640   20645640   T   C       exonic        PINK1   \n",
       "77218    chr1   20649109   20649109   C   T       exonic        PINK1   \n",
       "77228    chr1   20649217   20649217   C   T       exonic        PINK1   \n",
       "104335  chr20    5100921    5100921   C   A       exonic      TMEM230   \n",
       "113919  chr21   32726907   32726907   C   A       exonic        SYNJ1   \n",
       "114566  chr22   38115658   38115658   G   A       exonic       PLA2G6   \n",
       "116177  chr22   38132917   38132917   C   A       exonic       PLA2G6   \n",
       "119356  chr22   38169318   38169318   G   A       exonic       PLA2G6   \n",
       "346356   chr6  161350214  161350214   G   C     intronic         PRKN   \n",
       "382697   chr6  161785793  161785793   C   G       exonic         PRKN   \n",
       "382708   chr6  161785839  161785839   A   T       exonic         PRKN   \n",
       "440038   chr6  162443314  162443314   A   T       exonic         PRKN   \n",
       "440042   chr6  162443356  162443356   C   G       exonic         PRKN   \n",
       "\n",
       "                                       GeneDetail.refGene ExonicFunc.refGene  \\\n",
       "9898                                                    .  nonsynonymous SNV   \n",
       "47879                                                   .                  .   \n",
       "48145                                                   .  nonsynonymous SNV   \n",
       "48146                                                   .  nonsynonymous SNV   \n",
       "48530                                                   .  nonsynonymous SNV   \n",
       "48542                                                   .  nonsynonymous SNV   \n",
       "48682   NM_001126131:exon7:c.1433+1G>A;NM_002693:exon7...                  .   \n",
       "48685                                                   .  nonsynonymous SNV   \n",
       "48843                                                   .  nonsynonymous SNV   \n",
       "70751                                                   .  nonsynonymous SNV   \n",
       "70835                                                   .  nonsynonymous SNV   \n",
       "70948                                                   .  nonsynonymous SNV   \n",
       "71013                                                   .  nonsynonymous SNV   \n",
       "71017                                                   .  nonsynonymous SNV   \n",
       "71038                                                   .  nonsynonymous SNV   \n",
       "71134                                                   .  nonsynonymous SNV   \n",
       "72210                                                   .  nonsynonymous SNV   \n",
       "72328                                                   .           stopgain   \n",
       "72536                                                   .           stopgain   \n",
       "72904                                                   .           stopgain   \n",
       "73729                                                   .           stopgain   \n",
       "76909                                                   .  nonsynonymous SNV   \n",
       "77218                                                   .           stopgain   \n",
       "77228                                                   .           stopgain   \n",
       "104335                                                  .  nonsynonymous SNV   \n",
       "113919                                                  .           stopgain   \n",
       "114566                                                  .           stopgain   \n",
       "116177                                                  .  nonsynonymous SNV   \n",
       "119356                                                  .           stopgain   \n",
       "346356                                                  .                  .   \n",
       "382697                                                  .  nonsynonymous SNV   \n",
       "382708                                                  .           stopgain   \n",
       "440038                                                  .  nonsynonymous SNV   \n",
       "440042                                                  .  nonsynonymous SNV   \n",
       "\n",
       "                                         AAChange.refGene  ...  \\\n",
       "9898             LRRK2:NM_198578:exon31:c.C4321T:p.R1441C  ...   \n",
       "47879                                                   .  ...   \n",
       "48145   POLG:NM_001126131:exon16:c.C2554T:p.R852C,POLG...  ...   \n",
       "48146   POLG:NM_001126131:exon16:c.G2542A:p.G848S,POLG...  ...   \n",
       "48530   POLG:NM_001126131:exon10:c.C1789T:p.R597W,POLG...  ...   \n",
       "48542   POLG:NM_001126131:exon10:c.C1720T:p.R574W,POLG...  ...   \n",
       "48682                                                   .  ...   \n",
       "48685   POLG:NM_001126131:exon7:c.G1399A:p.A467T,POLG:...  ...   \n",
       "48843   POLG:NM_001126131:exon4:c.C925T:p.R309C,POLG:N...  ...   \n",
       "70751   GBA:NM_001171811:exon9:c.C1243T:p.R415C,GBA:NM...  ...   \n",
       "70835   GBA:NM_001171811:exon7:c.C913T:p.R305W,GBA:NM_...  ...   \n",
       "70948   GBA:NM_001171811:exon6:c.G626A:p.R209Q,GBA:NM_...  ...   \n",
       "71013   GBA:NM_001171811:exon5:c.T428G:p.V143G,GBA:NM_...  ...   \n",
       "71017   GBA:NM_001171811:exon5:c.A419G:p.N140S,GBA:NM_...  ...   \n",
       "71038   GBA:NM_001171811:exon4:c.C214T:p.R72W,GBA:NM_0...  ...   \n",
       "71134   GBA:NM_000157:exon3:c.C259T:p.R87W,GBA:NM_0011...  ...   \n",
       "72210   ATP13A2:NM_001141974:exon23:c.G2497A:p.G833R,A...  ...   \n",
       "72328   ATP13A2:NM_001141973:exon22:c.C2440T:p.R814X,A...  ...   \n",
       "72536   ATP13A2:NM_001141973:exon18:c.C1888T:p.Q630X,A...  ...   \n",
       "72904   ATP13A2:NM_001141973:exon15:c.C1444T:p.R482X,A...  ...   \n",
       "73729   ATP13A2:NM_001141973:exon3:c.G212A:p.W71X,ATP1...  ...   \n",
       "76909              PINK1:NM_032409:exon5:c.T1040C:p.L347P  ...   \n",
       "77218              PINK1:NM_032409:exon7:c.C1366T:p.Q456X  ...   \n",
       "77228              PINK1:NM_032409:exon7:c.C1474T:p.R492X  ...   \n",
       "104335  TMEM230:NM_001009925:exon4:c.G233T:p.R78L,TMEM...  ...   \n",
       "113919               SYNJ1:NM_003895:exon2:c.G106T:p.E36X  ...   \n",
       "114566  PLA2G6:NM_001004426:exon13:c.C1741T:p.R581X,PL...  ...   \n",
       "116177  PLA2G6:NM_001349868:exon6:c.G313T:p.D105Y,PLA2...  ...   \n",
       "119356  PLA2G6:NM_001004426:exon2:c.C109T:p.R37X,PLA2G...  ...   \n",
       "346356                                                  .  ...   \n",
       "382697  PRKN:NM_013988:exon4:c.G403C:p.G135R,PRKN:NM_0...  ...   \n",
       "382708  PRKN:NM_013988:exon4:c.T357A:p.C119X,PRKN:NM_0...  ...   \n",
       "440038  PRKN:NM_004562:exon2:c.T167A:p.V56E,PRKN:NM_01...  ...   \n",
       "440042  PRKN:NM_004562:exon2:c.G125C:p.R42P,PRKN:NM_01...  ...   \n",
       "\n",
       "                                               CLNREVSTAT      CLNSIG ONCDN  \\\n",
       "9898    criteria_provided,_multiple_submitters,_no_con...  Pathogenic     .   \n",
       "47879   criteria_provided,_multiple_submitters,_no_con...  Pathogenic     .   \n",
       "48145   criteria_provided,_multiple_submitters,_no_con...  Pathogenic     .   \n",
       "48146   criteria_provided,_multiple_submitters,_no_con...  Pathogenic     .   \n",
       "48530   criteria_provided,_multiple_submitters,_no_con...  Pathogenic     .   \n",
       "48542   criteria_provided,_multiple_submitters,_no_con...  Pathogenic     .   \n",
       "48682   criteria_provided,_multiple_submitters,_no_con...  Pathogenic     .   \n",
       "48685   criteria_provided,_multiple_submitters,_no_con...  Pathogenic     .   \n",
       "48843   criteria_provided,_multiple_submitters,_no_con...  Pathogenic     .   \n",
       "70751   criteria_provided,_multiple_submitters,_no_con...  Pathogenic     .   \n",
       "70835                 criteria_provided,_single_submitter  Pathogenic     .   \n",
       "70948   criteria_provided,_multiple_submitters,_no_con...  Pathogenic     .   \n",
       "71013                 criteria_provided,_single_submitter  Pathogenic     .   \n",
       "71017   criteria_provided,_multiple_submitters,_no_con...  Pathogenic     .   \n",
       "71038   criteria_provided,_multiple_submitters,_no_con...  Pathogenic     .   \n",
       "71134   criteria_provided,_multiple_submitters,_no_con...  Pathogenic     .   \n",
       "72210                      no_assertion_criteria_provided  Pathogenic     .   \n",
       "72328   criteria_provided,_multiple_submitters,_no_con...  Pathogenic     .   \n",
       "72536   criteria_provided,_multiple_submitters,_no_con...  Pathogenic     .   \n",
       "72904   criteria_provided,_multiple_submitters,_no_con...  Pathogenic     .   \n",
       "73729   criteria_provided,_multiple_submitters,_no_con...  Pathogenic     .   \n",
       "76909   criteria_provided,_multiple_submitters,_no_con...  Pathogenic     .   \n",
       "77218   criteria_provided,_multiple_submitters,_no_con...  Pathogenic     .   \n",
       "77228   criteria_provided,_multiple_submitters,_no_con...  Pathogenic     .   \n",
       "104335                criteria_provided,_single_submitter  Pathogenic     .   \n",
       "113919                criteria_provided,_single_submitter  Pathogenic     .   \n",
       "114566  criteria_provided,_multiple_submitters,_no_con...  Pathogenic     .   \n",
       "116177  criteria_provided,_multiple_submitters,_no_con...  Pathogenic     .   \n",
       "119356  criteria_provided,_multiple_submitters,_no_con...  Pathogenic     .   \n",
       "346356                criteria_provided,_single_submitter  Pathogenic     .   \n",
       "382697                criteria_provided,_single_submitter  Pathogenic     .   \n",
       "382708  criteria_provided,_multiple_submitters,_no_con...  Pathogenic     .   \n",
       "440038  criteria_provided,_multiple_submitters,_no_con...  Pathogenic     .   \n",
       "440042                criteria_provided,_single_submitter  Pathogenic     .   \n",
       "\n",
       "       ONCDISDB ONCREVSTAT ONC SCIDN SCIDISDB SCIREVSTAT SCI\\n  \n",
       "9898          .          .   .     .        .          .     .  \n",
       "47879         .          .   .     .        .          .     .  \n",
       "48145         .          .   .     .        .          .     .  \n",
       "48146         .          .   .     .        .          .     .  \n",
       "48530         .          .   .     .        .          .     .  \n",
       "48542         .          .   .     .        .          .     .  \n",
       "48682         .          .   .     .        .          .     .  \n",
       "48685         .          .   .     .        .          .     .  \n",
       "48843         .          .   .     .        .          .     .  \n",
       "70751         .          .   .     .        .          .     .  \n",
       "70835         .          .   .     .        .          .     .  \n",
       "70948         .          .   .     .        .          .     .  \n",
       "71013         .          .   .     .        .          .     .  \n",
       "71017         .          .   .     .        .          .     .  \n",
       "71038         .          .   .     .        .          .     .  \n",
       "71134         .          .   .     .        .          .     .  \n",
       "72210         .          .   .     .        .          .     .  \n",
       "72328         .          .   .     .        .          .     .  \n",
       "72536         .          .   .     .        .          .     .  \n",
       "72904         .          .   .     .        .          .     .  \n",
       "73729         .          .   .     .        .          .     .  \n",
       "76909         .          .   .     .        .          .     .  \n",
       "77218         .          .   .     .        .          .     .  \n",
       "77228         .          .   .     .        .          .     .  \n",
       "104335        .          .   .     .        .          .     .  \n",
       "113919        .          .   .     .        .          .     .  \n",
       "114566        .          .   .     .        .          .     .  \n",
       "116177        .          .   .     .        .          .     .  \n",
       "119356        .          .   .     .        .          .     .  \n",
       "346356        .          .   .     .        .          .     .  \n",
       "382697        .          .   .     .        .          .     .  \n",
       "382708        .          .   .     .        .          .     .  \n",
       "440038        .          .   .     .        .          .     .  \n",
       "440042        .          .   .     .        .          .     .  \n",
       "\n",
       "[34 rows x 24 columns]"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# extract only the variants that have a clinical significant == \"Pathogenic\"\n",
    "\n",
    "pathogenic_vars = annotated_select_genes[annotated_select_genes['CLNSIG']=='Pathogenic']\n",
    "\n",
    "# exclude the variants in POLG2 (not a gene of interest)\n",
    "pathogenic_vars = pathogenic_vars[~(pathogenic_vars['Gene.refGene']=='POLG2')]\n",
    "\n",
    "# save list of pathogenic variants to an output file\n",
    "pathogenic_vars.to_csv(f'{workspace}/annotated_pathogenic_vars.txt', sep='\\t', header=True, index=False)\n",
    "pathogenic_vars"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 77,
   "id": "f0fe62a1-0564-4226-9342-0328f471a6ba",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>Gene.refGene</th>\n",
       "      <th>avsnp151</th>\n",
       "      <th>CLNALLELEID</th>\n",
       "      <th>CLNDN</th>\n",
       "      <th>CLNDISDB</th>\n",
       "      <th>CLNREVSTAT</th>\n",
       "      <th>CLNSIG</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>9898</th>\n",
       "      <td>chr12</td>\n",
       "      <td>40310434</td>\n",
       "      <td>40310434</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>LRRK2</td>\n",
       "      <td>rs33939927</td>\n",
       "      <td>16977</td>\n",
       "      <td>not_provided|Autosomal_dominant_Parkinson_dise...</td>\n",
       "      <td>MedGen:C3661900|MONDO:MONDO:0011764,MedGen:C18...</td>\n",
       "      <td>criteria_provided,_multiple_submitters,_no_con...</td>\n",
       "      <td>Pathogenic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70751</th>\n",
       "      <td>chr1</td>\n",
       "      <td>155235196</td>\n",
       "      <td>155235196</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>GBA</td>\n",
       "      <td>rs80356771</td>\n",
       "      <td>19334</td>\n",
       "      <td>Parkinson_disease,_late-onset|Gaucher_disease-...</td>\n",
       "      <td>MONDO:MONDO:0008199,MedGen:C3160718,OMIM:16860...</td>\n",
       "      <td>criteria_provided,_multiple_submitters,_no_con...</td>\n",
       "      <td>Pathogenic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70948</th>\n",
       "      <td>chr1</td>\n",
       "      <td>155237453</td>\n",
       "      <td>155237453</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>GBA</td>\n",
       "      <td>rs78973108</td>\n",
       "      <td>19367</td>\n",
       "      <td>Parkinson_disease,_late-onset|Gaucher_disease-...</td>\n",
       "      <td>MONDO:MONDO:0008199,MedGen:C3160718,OMIM:16860...</td>\n",
       "      <td>criteria_provided,_multiple_submitters,_no_con...</td>\n",
       "      <td>Pathogenic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>71017</th>\n",
       "      <td>chr1</td>\n",
       "      <td>155238215</td>\n",
       "      <td>155238215</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>GBA</td>\n",
       "      <td>rs364897</td>\n",
       "      <td>19353</td>\n",
       "      <td>Gaucher_disease-ophthalmoplegia-cardiovascular...</td>\n",
       "      <td>MONDO:MONDO:0009268,MedGen:C1856476,OMIM:23100...</td>\n",
       "      <td>criteria_provided,_multiple_submitters,_no_con...</td>\n",
       "      <td>Pathogenic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>71134</th>\n",
       "      <td>chr1</td>\n",
       "      <td>155239934</td>\n",
       "      <td>155239934</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>GBA</td>\n",
       "      <td>rs1141814</td>\n",
       "      <td>19360</td>\n",
       "      <td>Parkinson_disease,_late-onset|Gaucher_disease-...</td>\n",
       "      <td>MONDO:MONDO:0008199,MedGen:C3160718,OMIM:16860...</td>\n",
       "      <td>criteria_provided,_multiple_submitters,_no_con...</td>\n",
       "      <td>Pathogenic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>76909</th>\n",
       "      <td>chr1</td>\n",
       "      <td>20645640</td>\n",
       "      <td>20645640</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>PINK1</td>\n",
       "      <td>rs28940285</td>\n",
       "      <td>17447</td>\n",
       "      <td>Autosomal_recessive_early-onset_Parkinson_dise...</td>\n",
       "      <td>MONDO:MONDO:0011613,MedGen:C1853833,OMIM:60590...</td>\n",
       "      <td>criteria_provided,_multiple_submitters,_no_con...</td>\n",
       "      <td>Pathogenic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>77218</th>\n",
       "      <td>chr1</td>\n",
       "      <td>20649109</td>\n",
       "      <td>20649109</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>PINK1</td>\n",
       "      <td>rs45539432</td>\n",
       "      <td>17454</td>\n",
       "      <td>not_provided|Autosomal_recessive_early-onset_P...</td>\n",
       "      <td>MedGen:C3661900|MONDO:MONDO:0011613,MedGen:C18...</td>\n",
       "      <td>criteria_provided,_multiple_submitters,_no_con...</td>\n",
       "      <td>Pathogenic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>77228</th>\n",
       "      <td>chr1</td>\n",
       "      <td>20649217</td>\n",
       "      <td>20649217</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>PINK1</td>\n",
       "      <td>rs34208370</td>\n",
       "      <td>425331</td>\n",
       "      <td>Autosomal_recessive_early-onset_Parkinson_dise...</td>\n",
       "      <td>MONDO:MONDO:0011613,MedGen:C1853833,OMIM:60590...</td>\n",
       "      <td>criteria_provided,_multiple_submitters,_no_con...</td>\n",
       "      <td>Pathogenic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>113919</th>\n",
       "      <td>chr21</td>\n",
       "      <td>32726907</td>\n",
       "      <td>32726907</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>SYNJ1</td>\n",
       "      <td>rs1040540690</td>\n",
       "      <td>1393547</td>\n",
       "      <td>Developmental_and_epileptic_encephalopathy,_53...</td>\n",
       "      <td>MONDO:MONDO:0033362,MedGen:C4479313,OMIM:61738...</td>\n",
       "      <td>criteria_provided,_single_submitter</td>\n",
       "      <td>Pathogenic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>116177</th>\n",
       "      <td>chr22</td>\n",
       "      <td>38132917</td>\n",
       "      <td>38132917</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>PLA2G6</td>\n",
       "      <td>rs199935023</td>\n",
       "      <td>39328</td>\n",
       "      <td>Infantile_neuroaxonal_dystrophy|Autosomal_rece...</td>\n",
       "      <td>MONDO:MONDO:0024457,MedGen:C0270724,OMIM:25660...</td>\n",
       "      <td>criteria_provided,_multiple_submitters,_no_con...</td>\n",
       "      <td>Pathogenic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>382697</th>\n",
       "      <td>chr6</td>\n",
       "      <td>161785793</td>\n",
       "      <td>161785793</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>PRKN</td>\n",
       "      <td>rs751037529</td>\n",
       "      <td>395419</td>\n",
       "      <td>Autism_spectrum_disorder|not_provided|Autosoma...</td>\n",
       "      <td>MONDO:MONDO:0005258,MeSH:D000067877,MedGen:C15...</td>\n",
       "      <td>criteria_provided,_single_submitter</td>\n",
       "      <td>Pathogenic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>440038</th>\n",
       "      <td>chr6</td>\n",
       "      <td>162443314</td>\n",
       "      <td>162443314</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>PRKN</td>\n",
       "      <td>rs137853059</td>\n",
       "      <td>22086</td>\n",
       "      <td>not_provided|Autosomal_recessive_juvenile_Park...</td>\n",
       "      <td>MedGen:CN517202|MONDO:MONDO:0010820,MedGen:C18...</td>\n",
       "      <td>criteria_provided,_multiple_submitters,_no_con...</td>\n",
       "      <td>Pathogenic</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "          Chr      Start        End Ref Alt Gene.refGene      avsnp151  \\\n",
       "9898    chr12   40310434   40310434   C   T        LRRK2    rs33939927   \n",
       "70751    chr1  155235196  155235196   G   A          GBA    rs80356771   \n",
       "70948    chr1  155237453  155237453   C   T          GBA    rs78973108   \n",
       "71017    chr1  155238215  155238215   T   C          GBA      rs364897   \n",
       "71134    chr1  155239934  155239934   G   A          GBA     rs1141814   \n",
       "76909    chr1   20645640   20645640   T   C        PINK1    rs28940285   \n",
       "77218    chr1   20649109   20649109   C   T        PINK1    rs45539432   \n",
       "77228    chr1   20649217   20649217   C   T        PINK1    rs34208370   \n",
       "113919  chr21   32726907   32726907   C   A        SYNJ1  rs1040540690   \n",
       "116177  chr22   38132917   38132917   C   A       PLA2G6   rs199935023   \n",
       "382697   chr6  161785793  161785793   C   G         PRKN   rs751037529   \n",
       "440038   chr6  162443314  162443314   A   T         PRKN   rs137853059   \n",
       "\n",
       "       CLNALLELEID                                              CLNDN  \\\n",
       "9898         16977  not_provided|Autosomal_dominant_Parkinson_dise...   \n",
       "70751        19334  Parkinson_disease,_late-onset|Gaucher_disease-...   \n",
       "70948        19367  Parkinson_disease,_late-onset|Gaucher_disease-...   \n",
       "71017        19353  Gaucher_disease-ophthalmoplegia-cardiovascular...   \n",
       "71134        19360  Parkinson_disease,_late-onset|Gaucher_disease-...   \n",
       "76909        17447  Autosomal_recessive_early-onset_Parkinson_dise...   \n",
       "77218        17454  not_provided|Autosomal_recessive_early-onset_P...   \n",
       "77228       425331  Autosomal_recessive_early-onset_Parkinson_dise...   \n",
       "113919     1393547  Developmental_and_epileptic_encephalopathy,_53...   \n",
       "116177       39328  Infantile_neuroaxonal_dystrophy|Autosomal_rece...   \n",
       "382697      395419  Autism_spectrum_disorder|not_provided|Autosoma...   \n",
       "440038       22086  not_provided|Autosomal_recessive_juvenile_Park...   \n",
       "\n",
       "                                                 CLNDISDB  \\\n",
       "9898    MedGen:C3661900|MONDO:MONDO:0011764,MedGen:C18...   \n",
       "70751   MONDO:MONDO:0008199,MedGen:C3160718,OMIM:16860...   \n",
       "70948   MONDO:MONDO:0008199,MedGen:C3160718,OMIM:16860...   \n",
       "71017   MONDO:MONDO:0009268,MedGen:C1856476,OMIM:23100...   \n",
       "71134   MONDO:MONDO:0008199,MedGen:C3160718,OMIM:16860...   \n",
       "76909   MONDO:MONDO:0011613,MedGen:C1853833,OMIM:60590...   \n",
       "77218   MedGen:C3661900|MONDO:MONDO:0011613,MedGen:C18...   \n",
       "77228   MONDO:MONDO:0011613,MedGen:C1853833,OMIM:60590...   \n",
       "113919  MONDO:MONDO:0033362,MedGen:C4479313,OMIM:61738...   \n",
       "116177  MONDO:MONDO:0024457,MedGen:C0270724,OMIM:25660...   \n",
       "382697  MONDO:MONDO:0005258,MeSH:D000067877,MedGen:C15...   \n",
       "440038  MedGen:CN517202|MONDO:MONDO:0010820,MedGen:C18...   \n",
       "\n",
       "                                               CLNREVSTAT      CLNSIG  \n",
       "9898    criteria_provided,_multiple_submitters,_no_con...  Pathogenic  \n",
       "70751   criteria_provided,_multiple_submitters,_no_con...  Pathogenic  \n",
       "70948   criteria_provided,_multiple_submitters,_no_con...  Pathogenic  \n",
       "71017   criteria_provided,_multiple_submitters,_no_con...  Pathogenic  \n",
       "71134   criteria_provided,_multiple_submitters,_no_con...  Pathogenic  \n",
       "76909   criteria_provided,_multiple_submitters,_no_con...  Pathogenic  \n",
       "77218   criteria_provided,_multiple_submitters,_no_con...  Pathogenic  \n",
       "77228   criteria_provided,_multiple_submitters,_no_con...  Pathogenic  \n",
       "113919                criteria_provided,_single_submitter  Pathogenic  \n",
       "116177  criteria_provided,_multiple_submitters,_no_con...  Pathogenic  \n",
       "382697                criteria_provided,_single_submitter  Pathogenic  \n",
       "440038  criteria_provided,_multiple_submitters,_no_con...  Pathogenic  "
      ]
     },
     "execution_count": 77,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# check how many of the pathogenic variants have the term \"Parkinson\" in ClinVar's 'CLNDN' field\n",
    "\n",
    "pd_pathogenic_vars = pathogenic_vars[pathogenic_vars['CLNDN'].str.contains(\"parkinson\", case=False, regex=False)]\n",
    "pd_pathogenic_vars"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "ec66cff6-5ca0-420c-9dc1-07593cb6b922",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GBA CLNSIG\n",
      ".                                                           107049\n",
      "Uncertain_significance                                          43\n",
      "Likely_benign                                                   17\n",
      "Conflicting_classifications_of_pathogenicity                    13\n",
      "Pathogenic                                                       7\n",
      "Pathogenic/Likely_pathogenic                                     6\n",
      "Likely_pathogenic                                                3\n",
      "Benign                                                           3\n",
      "Benign/Likely_benign                                             2\n",
      "Pathogenic/Likely_pathogenic|risk_factor                         1\n",
      "Conflicting_classifications_of_pathogenicity|risk_factor         1\n",
      "Conflicting_classifications_of_pathogenicity|other               1\n",
      "Name: count, dtype: int64 \n",
      "\n",
      "ATP13A2 CLNSIG\n",
      ".                                               2805\n",
      "Likely_benign                                    157\n",
      "Uncertain_significance                           125\n",
      "Conflicting_classifications_of_pathogenicity      75\n",
      "Benign                                            30\n",
      "Benign/Likely_benign                              25\n",
      "Pathogenic                                         5\n",
      "Pathogenic/Likely_pathogenic                       2\n",
      "Likely_pathogenic                                  1\n",
      "Name: count, dtype: int64 \n",
      "\n",
      "PINK1 CLNSIG\n",
      ".                                               2519\n",
      "Uncertain_significance                            82\n",
      "Likely_benign                                     45\n",
      "Conflicting_classifications_of_pathogenicity      27\n",
      "Benign                                            27\n",
      "Benign/Likely_benign                              14\n",
      "Pathogenic                                         3\n",
      "Pathogenic/Likely_pathogenic                       2\n",
      "Uncertain_risk_allele                              1\n",
      "Name: count, dtype: int64 \n",
      "\n",
      "LRRK2 CLNSIG\n",
      ".                                                           15330\n",
      "Likely_benign                                                 183\n",
      "Uncertain_significance                                        141\n",
      "Benign                                                        128\n",
      "Benign/Likely_benign                                           43\n",
      "Conflicting_classifications_of_pathogenicity                   39\n",
      "not_provided                                                    2\n",
      "Pathogenic                                                      1\n",
      "Pathogenic/Likely_pathogenic|risk_factor                        1\n",
      "Conflicting_classifications_of_pathogenicity|risk_factor        1\n",
      "Name: count, dtype: int64 \n",
      "\n",
      "VPS13C CLNSIG\n",
      ".                                               28405\n",
      "Benign                                            171\n",
      "Likely_benign                                     149\n",
      "Uncertain_significance                            131\n",
      "Benign/Likely_benign                               14\n",
      "Conflicting_classifications_of_pathogenicity        9\n",
      "Likely_pathogenic                                   6\n",
      "Pathogenic/Likely_pathogenic                        1\n",
      "Name: count, dtype: int64 \n",
      "\n",
      "POLG CLNSIG\n",
      ".                                               31956\n",
      "Uncertain_significance                            203\n",
      "Likely_benign                                     165\n",
      "Conflicting_classifications_of_pathogenicity      142\n",
      "Benign                                             67\n",
      "Benign/Likely_benign                               32\n",
      "Pathogenic/Likely_pathogenic                       10\n",
      "Pathogenic                                          9\n",
      "Likely_pathogenic                                   2\n",
      "Name: count, dtype: int64 \n",
      "\n",
      "VPS35 CLNSIG\n",
      ".                                               15315\n",
      "Likely_benign                                      27\n",
      "Uncertain_significance                             25\n",
      "Benign                                             22\n",
      "Benign/Likely_benign                                6\n",
      "Conflicting_classifications_of_pathogenicity        4\n",
      "Name: count, dtype: int64 \n",
      "\n",
      "DNAJC6 CLNSIG\n",
      ".                                               13300\n",
      "Uncertain_significance                             57\n",
      "Likely_benign                                      53\n",
      "Benign                                             40\n",
      "Benign/Likely_benign                                9\n",
      "Conflicting_classifications_of_pathogenicity        2\n",
      "not_provided                                        1\n",
      "Name: count, dtype: int64 \n",
      "\n",
      "PARK7 CLNSIG\n",
      ".                                               5786\n",
      "Benign                                            25\n",
      "Uncertain_significance                            21\n",
      "Likely_benign                                     10\n",
      "Benign/Likely_benign                               4\n",
      "Likely_pathogenic                                  1\n",
      "Conflicting_classifications_of_pathogenicity       1\n",
      "Name: count, dtype: int64 \n",
      "\n",
      "SYNJ1 CLNSIG\n",
      ".                                               8167\n",
      "Likely_benign                                    161\n",
      "Uncertain_significance                           140\n",
      "Benign                                            33\n",
      "Conflicting_classifications_of_pathogenicity      12\n",
      "Benign/Likely_benign                               9\n",
      "Pathogenic                                         1\n",
      "Name: count, dtype: int64 \n",
      "\n",
      "FBXO7 Series([], Name: count, dtype: int64) \n",
      "\n",
      "PLA2G6 CLNSIG\n",
      ".                                               7765\n",
      "Likely_benign                                    110\n",
      "Uncertain_significance                            94\n",
      "Benign                                            50\n",
      "Conflicting_classifications_of_pathogenicity      46\n",
      "Benign/Likely_benign                               9\n",
      "Pathogenic                                         3\n",
      "Likely_pathogenic                                  1\n",
      "Pathogenic/Likely_pathogenic                       1\n",
      "Name: count, dtype: int64 \n",
      "\n",
      "HTRA2 CLNSIG\n",
      ".                                               173\n",
      "Likely_benign                                    29\n",
      "Uncertain_significance                           25\n",
      "Benign                                           10\n",
      "Benign/Likely_benign                              5\n",
      "Conflicting_classifications_of_pathogenicity      5\n",
      "Name: count, dtype: int64 \n",
      "\n",
      "DNAJC13 CLNSIG\n",
      ".                                               13709\n",
      "Benign                                             96\n",
      "Uncertain_significance                             39\n",
      "Likely_benign                                      18\n",
      "Benign/Likely_benign                                5\n",
      "Conflicting_classifications_of_pathogenicity        1\n",
      "Name: count, dtype: int64 \n",
      "\n",
      "UCHL1 CLNSIG\n",
      ".                                               11441\n",
      "Likely_benign                                      26\n",
      "Benign                                             21\n",
      "Uncertain_significance                             16\n",
      "Benign/Likely_benign                                4\n",
      "Conflicting_classifications_of_pathogenicity        3\n",
      "Name: count, dtype: int64 \n",
      "\n",
      "SNCA CLNSIG\n",
      ".                                               44894\n",
      "Uncertain_significance                             25\n",
      "Likely_benign                                      25\n",
      "Benign                                             19\n",
      "Benign/Likely_benign                                5\n",
      "Conflicting_classifications_of_pathogenicity        3\n",
      "Name: count, dtype: int64 \n",
      "\n",
      "PRKN CLNSIG\n",
      ".                                               124277\n",
      "Uncertain_significance                              68\n",
      "Likely_benign                                       60\n",
      "Benign                                              41\n",
      "Conflicting_classifications_of_pathogenicity        18\n",
      "Benign/Likely_benign                                12\n",
      "Pathogenic                                           5\n",
      "Pathogenic/Likely_pathogenic                         4\n",
      "Name: count, dtype: int64 \n",
      "\n",
      "GIGYF2 CLNSIG\n",
      ".                                               14086\n",
      "Benign                                             38\n",
      "Uncertain_significance                             23\n",
      "Likely_benign                                      15\n",
      "risk_factor                                         3\n",
      "Conflicting_classifications_of_pathogenicity        1\n",
      "Benign/Likely_benign                                1\n",
      "Name: count, dtype: int64 \n",
      "\n",
      "EIF4G1 CLNSIG\n",
      ".                         2185\n",
      "Uncertain_significance      28\n",
      "Benign                      26\n",
      "Likely_benign               21\n",
      "Benign/Likely_benign         3\n",
      "risk_factor                  2\n",
      "Name: count, dtype: int64 \n",
      "\n",
      "TMEM230 CLNSIG\n",
      ".                                               8630\n",
      "Likely_benign                                      9\n",
      "Benign                                             6\n",
      "Uncertain_significance                             6\n",
      "Pathogenic                                         1\n",
      "Conflicting_classifications_of_pathogenicity       1\n",
      "Name: count, dtype: int64 \n",
      "\n",
      "LRP10 CLNSIG\n",
      ".                         2817\n",
      "Uncertain_significance      18\n",
      "Benign                      11\n",
      "Likely_benign                9\n",
      "Name: count, dtype: int64 \n",
      "\n"
     ]
    }
   ],
   "source": [
    "# check the clinical significance counts for all PD genes of interest\n",
    "\n",
    "for gene in ['GBA','ATP13A2','PINK1','LRRK2','VPS13C','POLG','VPS35','DNAJC6','PARK7','SYNJ1', \\\n",
    "             'FBXO7','PLA2G6','HTRA2','DNAJC13','UCHL1','SNCA','PRKN','GIGYF2','EIF4G1','TMEM230','LRP10']:\n",
    "    path_vars_gene = annotated_select_genes[(annotated_select_genes['Gene.refGene'].str.contains(gene))]\n",
    "    print(gene, path_vars_gene.CLNSIG.value_counts(), '\\n')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 81,
   "id": "7ad6ee83-dfe3-448f-8a5c-d8e5480c44bd",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# save just variant IDs to calculate freqs from\n",
    "\n",
    "annotated_pathogenic_vars = pd.read_csv(f'{workspace}/annotated_pathogenic_vars.txt', sep='\\t')\n",
    "annotated_pathogenic_vars['snpID'] = annotated_pathogenic_vars['Chr'] + ':' \\\n",
    "                                     + annotated_pathogenic_vars['Start'].astype(str) + ':' \\\n",
    "                                     + annotated_pathogenic_vars['Ref'] + ':' \\\n",
    "                                     + annotated_pathogenic_vars['Alt']\n",
    "annotated_pathogenic_vars['snpID'].to_csv(f'{workspace}/pathogenic_variant_IDs.txt', header=None, index=None)                                    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 82,
   "id": "bc6e5833-d1de-44a0-89a7-2806692fe9cf",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/bfiles/AAC_r7_pathogenic_only.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/workspace/bfiles/MERGE_AAC\n",
      "  --extract /home/jupyter/workspace/pathogenic_variant_IDs.txt\n",
      "  --make-bed\n",
      "  --out /home/jupyter/workspace/bfiles/AAC_r7_pathogenic_only\n",
      "\n",
      "209368 MB RAM detected; reserving 104684 MB for main workspace.\n",
      "34460677 variants loaded from .bim file.\n",
      "1111 people (455 males, 656 females) loaded from .fam.\n",
      "1086 phenotype values loaded from .fam.\n",
      "--extract: 2 variants remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 1111 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.99955.\n",
      "2 variants and 1111 people pass filters and QC.\n",
      "Among remaining phenotypes, 285 are cases and 801 are controls.  (25 phenotypes\n",
      "are missing.)\n",
      "--make-bed to /home/jupyter/workspace/bfiles/AAC_r7_pathogenic_only.bed +\n",
      "/home/jupyter/workspace/bfiles/AAC_r7_pathogenic_only.bim +\n",
      "/home/jupyter/workspace/bfiles/AAC_r7_pathogenic_only.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/bfiles/AFR_r7_pathogenic_only.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/workspace/bfiles/MERGE_AFR\n",
      "  --extract /home/jupyter/workspace/pathogenic_variant_IDs.txt\n",
      "  --make-bed\n",
      "  --out /home/jupyter/workspace/bfiles/AFR_r7_pathogenic_only\n",
      "\n",
      "209368 MB RAM detected; reserving 104684 MB for main workspace.\n",
      "39494539 variants loaded from .bim file.\n",
      "2643 people (1480 males, 1163 females) loaded from .fam.\n",
      "2621 phenotype values loaded from .fam.\n",
      "--extract: 8 variants remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 2643 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999054.\n",
      "8 variants and 2643 people pass filters and QC.\n",
      "Among remaining phenotypes, 942 are cases and 1679 are controls.  (22\n",
      "phenotypes are missing.)\n",
      "--make-bed to /home/jupyter/workspace/bfiles/AFR_r7_pathogenic_only.bed +\n",
      "/home/jupyter/workspace/bfiles/AFR_r7_pathogenic_only.bim +\n",
      "/home/jupyter/workspace/bfiles/AFR_r7_pathogenic_only.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/bfiles/AJ_r7_pathogenic_only.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/workspace/bfiles/MERGE_AJ\n",
      "  --extract /home/jupyter/workspace/pathogenic_variant_IDs.txt\n",
      "  --make-bed\n",
      "  --out /home/jupyter/workspace/bfiles/AJ_r7_pathogenic_only\n",
      "\n",
      "209368 MB RAM detected; reserving 104684 MB for main workspace.\n",
      "14332784 variants loaded from .bim file.\n",
      "2655 people (1648 males, 1007 females) loaded from .fam.\n",
      "1703 phenotype values loaded from .fam.\n",
      "--extract: 2 variants remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 2655 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999812.\n",
      "2 variants and 2655 people pass filters and QC.\n",
      "Among remaining phenotypes, 1292 are cases and 411 are controls.  (952\n",
      "phenotypes are missing.)\n",
      "--make-bed to /home/jupyter/workspace/bfiles/AJ_r7_pathogenic_only.bed +\n",
      "/home/jupyter/workspace/bfiles/AJ_r7_pathogenic_only.bim +\n",
      "/home/jupyter/workspace/bfiles/AJ_r7_pathogenic_only.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/bfiles/AMR_r7_pathogenic_only.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/workspace/bfiles/MERGE_AMR\n",
      "  --extract /home/jupyter/workspace/pathogenic_variant_IDs.txt\n",
      "  --make-bed\n",
      "  --out /home/jupyter/workspace/bfiles/AMR_r7_pathogenic_only\n",
      "\n",
      "209368 MB RAM detected; reserving 104684 MB for main workspace.\n",
      "20873391 variants loaded from .bim file.\n",
      "646 people (363 males, 283 females) loaded from .fam.\n",
      "613 phenotype values loaded from .fam.\n",
      "--extract: 3 variants remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 646 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999484.\n",
      "3 variants and 646 people pass filters and QC.\n",
      "Among remaining phenotypes, 458 are cases and 155 are controls.  (33 phenotypes\n",
      "are missing.)\n",
      "--make-bed to /home/jupyter/workspace/bfiles/AMR_r7_pathogenic_only.bed +\n",
      "/home/jupyter/workspace/bfiles/AMR_r7_pathogenic_only.bim +\n",
      "/home/jupyter/workspace/bfiles/AMR_r7_pathogenic_only.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/bfiles/CAH_r7_pathogenic_only.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/workspace/bfiles/MERGE_CAH\n",
      "  --extract /home/jupyter/workspace/pathogenic_variant_IDs.txt\n",
      "  --make-bed\n",
      "  --out /home/jupyter/workspace/bfiles/CAH_r7_pathogenic_only\n",
      "\n",
      "209368 MB RAM detected; reserving 104684 MB for main workspace.\n",
      "31670734 variants loaded from .bim file.\n",
      "851 people (446 males, 405 females) loaded from .fam.\n",
      "827 phenotype values loaded from .fam.\n",
      "--extract: 2 variants remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 851 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "2 variants and 851 people pass filters and QC.\n",
      "Among remaining phenotypes, 525 are cases and 302 are controls.  (24 phenotypes\n",
      "are missing.)\n",
      "--make-bed to /home/jupyter/workspace/bfiles/CAH_r7_pathogenic_only.bed +\n",
      "/home/jupyter/workspace/bfiles/CAH_r7_pathogenic_only.bim +\n",
      "/home/jupyter/workspace/bfiles/CAH_r7_pathogenic_only.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/bfiles/CAS_r7_pathogenic_only.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/workspace/bfiles/MERGE_CAS\n",
      "  --extract /home/jupyter/workspace/pathogenic_variant_IDs.txt\n",
      "  --make-bed\n",
      "  --out /home/jupyter/workspace/bfiles/CAS_r7_pathogenic_only\n",
      "\n",
      "209368 MB RAM detected; reserving 104684 MB for main workspace.\n",
      "19008457 variants loaded from .bim file.\n",
      "903 people (439 males, 464 females) loaded from .fam.\n",
      "895 phenotype values loaded from .fam.\n",
      "--extract: 2 variants remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 903 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "2 variants and 903 people pass filters and QC.\n",
      "Among remaining phenotypes, 552 are cases and 343 are controls.  (8 phenotypes\n",
      "are missing.)\n",
      "--make-bed to /home/jupyter/workspace/bfiles/CAS_r7_pathogenic_only.bed +\n",
      "/home/jupyter/workspace/bfiles/CAS_r7_pathogenic_only.bim +\n",
      "/home/jupyter/workspace/bfiles/CAS_r7_pathogenic_only.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/bfiles/EAS_r7_pathogenic_only.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/workspace/bfiles/MERGE_EAS\n",
      "  --extract /home/jupyter/workspace/pathogenic_variant_IDs.txt\n",
      "  --make-bed\n",
      "  --out /home/jupyter/workspace/bfiles/EAS_r7_pathogenic_only\n",
      "\n",
      "209368 MB RAM detected; reserving 104684 MB for main workspace.\n",
      "28698631 variants loaded from .bim file.\n",
      "5167 people (3278 males, 1889 females) loaded from .fam.\n",
      "5123 phenotype values loaded from .fam.\n",
      "--extract: 5 variants remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 5167 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.998916.\n",
      "5 variants and 5167 people pass filters and QC.\n",
      "Among remaining phenotypes, 2662 are cases and 2461 are controls.  (44\n",
      "phenotypes are missing.)\n",
      "--make-bed to /home/jupyter/workspace/bfiles/EAS_r7_pathogenic_only.bed +\n",
      "/home/jupyter/workspace/bfiles/EAS_r7_pathogenic_only.bim +\n",
      "/home/jupyter/workspace/bfiles/EAS_r7_pathogenic_only.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/bfiles/EUR_r7_pathogenic_only.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/workspace/bfiles/MERGE_EUR\n",
      "  --extract /home/jupyter/workspace/pathogenic_variant_IDs.txt\n",
      "  --make-bed\n",
      "  --out /home/jupyter/workspace/bfiles/EUR_r7_pathogenic_only\n",
      "\n",
      "209368 MB RAM detected; reserving 104684 MB for main workspace.\n",
      "91242179 variants loaded from .bim file.\n",
      "38839 people (23296 males, 15543 females) loaded from .fam.\n",
      "30412 phenotype values loaded from .fam.\n",
      "--extract: 19 variants remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 38839 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.841842.\n",
      "19 variants and 38839 people pass filters and QC.\n",
      "Among remaining phenotypes, 21198 are cases and 9214 are controls.  (8427\n",
      "phenotypes are missing.)\n",
      "--make-bed to /home/jupyter/workspace/bfiles/EUR_r7_pathogenic_only.bed +\n",
      "/home/jupyter/workspace/bfiles/EUR_r7_pathogenic_only.bim +\n",
      "/home/jupyter/workspace/bfiles/EUR_r7_pathogenic_only.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/bfiles/FIN_r7_pathogenic_only.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/workspace/bfiles/MERGE_FIN\n",
      "  --extract /home/jupyter/workspace/pathogenic_variant_IDs.txt\n",
      "  --make-bed\n",
      "  --out /home/jupyter/workspace/bfiles/FIN_r7_pathogenic_only\n",
      "\n",
      "209368 MB RAM detected; reserving 104684 MB for main workspace.\n",
      "7522443 variants loaded from .bim file.\n",
      "114 people (53 males, 61 females) loaded from .fam.\n",
      "106 phenotype values loaded from .fam.\n",
      "--extract: 1 variant remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 114 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.991228.\n",
      "1 variant and 114 people pass filters and QC.\n",
      "Among remaining phenotypes, 98 are cases and 8 are controls.  (8 phenotypes are\n",
      "missing.)\n",
      "--make-bed to /home/jupyter/workspace/bfiles/FIN_r7_pathogenic_only.bed +\n",
      "/home/jupyter/workspace/bfiles/FIN_r7_pathogenic_only.bim +\n",
      "/home/jupyter/workspace/bfiles/FIN_r7_pathogenic_only.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/bfiles/MDE_r7_pathogenic_only.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/workspace/bfiles/MERGE_MDE\n",
      "  --extract /home/jupyter/workspace/pathogenic_variant_IDs.txt\n",
      "  --make-bed\n",
      "  --out /home/jupyter/workspace/bfiles/MDE_r7_pathogenic_only\n",
      "\n",
      "209368 MB RAM detected; reserving 104684 MB for main workspace.\n",
      "18471227 variants loaded from .bim file.\n",
      "581 people (369 males, 212 females) loaded from .fam.\n",
      "536 phenotype values loaded from .fam.\n",
      "--extract: 3 variants remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 581 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999426.\n",
      "3 variants and 581 people pass filters and QC.\n",
      "Among remaining phenotypes, 311 are cases and 225 are controls.  (45 phenotypes\n",
      "are missing.)\n",
      "--make-bed to /home/jupyter/workspace/bfiles/MDE_r7_pathogenic_only.bed +\n",
      "/home/jupyter/workspace/bfiles/MDE_r7_pathogenic_only.bim +\n",
      "/home/jupyter/workspace/bfiles/MDE_r7_pathogenic_only.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/bfiles/SAS_r7_pathogenic_only.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/workspace/bfiles/MERGE_SAS\n",
      "  --extract /home/jupyter/workspace/pathogenic_variant_IDs.txt\n",
      "  --make-bed\n",
      "  --out /home/jupyter/workspace/bfiles/SAS_r7_pathogenic_only\n",
      "\n",
      "209368 MB RAM detected; reserving 104684 MB for main workspace.\n",
      "16746115 variants loaded from .bim file.\n",
      "635 people (408 males, 227 females) loaded from .fam.\n",
      "609 phenotype values loaded from .fam.\n",
      "Error: No variants remaining after --extract.\n"
     ]
    }
   ],
   "source": [
    "# extract pathogenic variants from each ancestry plink files\n",
    "\n",
    "for anc in ancestries:\n",
    "    plink_cmd = f'{plink_exec} --bfile {workspace}/bfiles/MERGE_{anc} --extract {workspace}/pathogenic_variant_IDs.txt --make-bed --out {workspace}/bfiles/{anc}_r7_pathogenic_only'\n",
    "    !{plink_cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 83,
   "id": "b1ef6abb-f24a-42f0-bdfa-ccb185e90fec",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "AAC (2, 6)\n",
      "AFR (8, 6)\n",
      "AJ (2, 6)\n",
      "AMR (3, 6)\n",
      "CAH (2, 6)\n",
      "CAS (2, 6)\n",
      "EAS (5, 6)\n",
      "EUR (19, 6)\n",
      "FIN (1, 6)\n",
      "MDE (3, 6)\n",
      "SAS no pathogenic vars\n"
     ]
    }
   ],
   "source": [
    "# check how many pathogenic variants in each ancestry\n",
    "\n",
    "for anc in ancestries:\n",
    "    bim_fn = f'{workspace}/bfiles/{anc}_r7_pathogenic_only.bim'\n",
    "    if os.path.isfile(bim_fn):\n",
    "        bim = pd.read_csv(bim_fn, sep='\\s+', header=None, names=['CHR','snpID','DIST','POS','ALT','REF'])\n",
    "        print(anc, bim.shape)\n",
    "    else:\n",
    "        print(anc, \"no pathogenic vars\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 84,
   "id": "4bd7f391-bc73-48de-a9f3-f1215fe3dd6f",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/freqs/AAC_path_vars_case_control.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/workspace/bfiles/AAC_r7_pathogenic_only\n",
      "  --freq case-control\n",
      "  --out /home/jupyter/workspace/freqs/AAC_path_vars_case_control\n",
      "\n",
      "209368 MB RAM detected; reserving 104684 MB for main workspace.\n",
      "2 variants loaded from .bim file.\n",
      "1111 people (455 males, 656 females) loaded from .fam.\n",
      "1086 phenotype values loaded from .fam.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 1111 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.99955.\n",
      "--freq case-control: Phenotype-stratified allele frequencies (founders only)\n",
      "written to /home/jupyter/workspace/freqs/AAC_path_vars_case_control.frq.cc .\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/freqs/AFR_path_vars_case_control.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/workspace/bfiles/AFR_r7_pathogenic_only\n",
      "  --freq case-control\n",
      "  --out /home/jupyter/workspace/freqs/AFR_path_vars_case_control\n",
      "\n",
      "209368 MB RAM detected; reserving 104684 MB for main workspace.\n",
      "8 variants loaded from .bim file.\n",
      "2643 people (1480 males, 1163 females) loaded from .fam.\n",
      "2621 phenotype values loaded from .fam.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 2643 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999054.\n",
      "--freq case-control: Phenotype-stratified allele frequencies (founders only)\n",
      "written to /home/jupyter/workspace/freqs/AFR_path_vars_case_control.frq.cc .\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/freqs/AJ_path_vars_case_control.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/workspace/bfiles/AJ_r7_pathogenic_only\n",
      "  --freq case-control\n",
      "  --out /home/jupyter/workspace/freqs/AJ_path_vars_case_control\n",
      "\n",
      "209368 MB RAM detected; reserving 104684 MB for main workspace.\n",
      "2 variants loaded from .bim file.\n",
      "2655 people (1648 males, 1007 females) loaded from .fam.\n",
      "1703 phenotype values loaded from .fam.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 2655 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999812.\n",
      "--freq case-control: Phenotype-stratified allele frequencies (founders only)\n",
      "written to /home/jupyter/workspace/freqs/AJ_path_vars_case_control.frq.cc .\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/freqs/AMR_path_vars_case_control.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/workspace/bfiles/AMR_r7_pathogenic_only\n",
      "  --freq case-control\n",
      "  --out /home/jupyter/workspace/freqs/AMR_path_vars_case_control\n",
      "\n",
      "209368 MB RAM detected; reserving 104684 MB for main workspace.\n",
      "3 variants loaded from .bim file.\n",
      "646 people (363 males, 283 females) loaded from .fam.\n",
      "613 phenotype values loaded from .fam.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 646 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999484.\n",
      "--freq case-control: Phenotype-stratified allele frequencies (founders only)\n",
      "written to /home/jupyter/workspace/freqs/AMR_path_vars_case_control.frq.cc .\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/freqs/CAH_path_vars_case_control.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/workspace/bfiles/CAH_r7_pathogenic_only\n",
      "  --freq case-control\n",
      "  --out /home/jupyter/workspace/freqs/CAH_path_vars_case_control\n",
      "\n",
      "209368 MB RAM detected; reserving 104684 MB for main workspace.\n",
      "2 variants loaded from .bim file.\n",
      "851 people (446 males, 405 females) loaded from .fam.\n",
      "827 phenotype values loaded from .fam.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 851 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "--freq case-control: Phenotype-stratified allele frequencies (founders only)\n",
      "written to /home/jupyter/workspace/freqs/CAH_path_vars_case_control.frq.cc .\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/freqs/CAS_path_vars_case_control.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/workspace/bfiles/CAS_r7_pathogenic_only\n",
      "  --freq case-control\n",
      "  --out /home/jupyter/workspace/freqs/CAS_path_vars_case_control\n",
      "\n",
      "209368 MB RAM detected; reserving 104684 MB for main workspace.\n",
      "2 variants loaded from .bim file.\n",
      "903 people (439 males, 464 females) loaded from .fam.\n",
      "895 phenotype values loaded from .fam.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 903 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "--freq case-control: Phenotype-stratified allele frequencies (founders only)\n",
      "written to /home/jupyter/workspace/freqs/CAS_path_vars_case_control.frq.cc .\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/freqs/EAS_path_vars_case_control.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/workspace/bfiles/EAS_r7_pathogenic_only\n",
      "  --freq case-control\n",
      "  --out /home/jupyter/workspace/freqs/EAS_path_vars_case_control\n",
      "\n",
      "209368 MB RAM detected; reserving 104684 MB for main workspace.\n",
      "5 variants loaded from .bim file.\n",
      "5167 people (3278 males, 1889 females) loaded from .fam.\n",
      "5123 phenotype values loaded from .fam.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 5167 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.998916.\n",
      "--freq case-control: Phenotype-stratified allele frequencies (founders only)\n",
      "written to /home/jupyter/workspace/freqs/EAS_path_vars_case_control.frq.cc .\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/freqs/EUR_path_vars_case_control.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/workspace/bfiles/EUR_r7_pathogenic_only\n",
      "  --freq case-control\n",
      "  --out /home/jupyter/workspace/freqs/EUR_path_vars_case_control\n",
      "\n",
      "209368 MB RAM detected; reserving 104684 MB for main workspace.\n",
      "19 variants loaded from .bim file.\n",
      "38839 people (23296 males, 15543 females) loaded from .fam.\n",
      "30412 phenotype values loaded from .fam.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 38839 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.841842.\n",
      "--freq case-control: Phenotype-stratified allele frequencies (founders only)\n",
      "written to /home/jupyter/workspace/freqs/EUR_path_vars_case_control.frq.cc .\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/freqs/FIN_path_vars_case_control.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/workspace/bfiles/FIN_r7_pathogenic_only\n",
      "  --freq case-control\n",
      "  --out /home/jupyter/workspace/freqs/FIN_path_vars_case_control\n",
      "\n",
      "209368 MB RAM detected; reserving 104684 MB for main workspace.\n",
      "1 variant loaded from .bim file.\n",
      "114 people (53 males, 61 females) loaded from .fam.\n",
      "106 phenotype values loaded from .fam.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 114 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.991228.\n",
      "--freq case-control: Phenotype-stratified allele frequencies (founders only)\n",
      "written to /home/jupyter/workspace/freqs/FIN_path_vars_case_control.frq.cc .\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/freqs/MDE_path_vars_case_control.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/workspace/bfiles/MDE_r7_pathogenic_only\n",
      "  --freq case-control\n",
      "  --out /home/jupyter/workspace/freqs/MDE_path_vars_case_control\n",
      "\n",
      "209368 MB RAM detected; reserving 104684 MB for main workspace.\n",
      "3 variants loaded from .bim file.\n",
      "581 people (369 males, 212 females) loaded from .fam.\n",
      "536 phenotype values loaded from .fam.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 581 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999426.\n",
      "--freq case-control: Phenotype-stratified allele frequencies (founders only)\n",
      "written to /home/jupyter/workspace/freqs/MDE_path_vars_case_control.frq.cc .\n"
     ]
    }
   ],
   "source": [
    "# run frequencies for case/control per ancestry\n",
    "\n",
    "for anc in ancestries:\n",
    "    anc_path = f'{workspace}/bfiles/{anc}_r7_pathogenic_only'\n",
    "    if os.path.isfile(f'{anc_path}.bim'):\n",
    "        plink_cmd = f'{plink_exec} --bfile {anc_path} --freq case-control --out {workspace}/freqs/{anc}_path_vars_case_control'\n",
    "        !{plink_cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 85,
   "id": "6c2d7494-45cb-41fc-8fe2-24ba976dbbd5",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/freqs/AAC_path_vars_all.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/workspace/bfiles/AAC_r7_pathogenic_only\n",
      "  --freq\n",
      "  --out /home/jupyter/workspace/freqs/AAC_path_vars_all\n",
      "\n",
      "209368 MB RAM detected; reserving 104684 MB for main workspace.\n",
      "2 variants loaded from .bim file.\n",
      "1111 people (455 males, 656 females) loaded from .fam.\n",
      "1086 phenotype values loaded from .fam.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 1111 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.99955.\n",
      "--freq: Allele frequencies (founders only) written to\n",
      "/home/jupyter/workspace/freqs/AAC_path_vars_all.frq .\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/freqs/AFR_path_vars_all.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/workspace/bfiles/AFR_r7_pathogenic_only\n",
      "  --freq\n",
      "  --out /home/jupyter/workspace/freqs/AFR_path_vars_all\n",
      "\n",
      "209368 MB RAM detected; reserving 104684 MB for main workspace.\n",
      "8 variants loaded from .bim file.\n",
      "2643 people (1480 males, 1163 females) loaded from .fam.\n",
      "2621 phenotype values loaded from .fam.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 2643 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999054.\n",
      "--freq: Allele frequencies (founders only) written to\n",
      "/home/jupyter/workspace/freqs/AFR_path_vars_all.frq .\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/freqs/AJ_path_vars_all.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/workspace/bfiles/AJ_r7_pathogenic_only\n",
      "  --freq\n",
      "  --out /home/jupyter/workspace/freqs/AJ_path_vars_all\n",
      "\n",
      "209368 MB RAM detected; reserving 104684 MB for main workspace.\n",
      "2 variants loaded from .bim file.\n",
      "2655 people (1648 males, 1007 females) loaded from .fam.\n",
      "1703 phenotype values loaded from .fam.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 2655 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999812.\n",
      "--freq: Allele frequencies (founders only) written to\n",
      "/home/jupyter/workspace/freqs/AJ_path_vars_all.frq .\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/freqs/AMR_path_vars_all.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/workspace/bfiles/AMR_r7_pathogenic_only\n",
      "  --freq\n",
      "  --out /home/jupyter/workspace/freqs/AMR_path_vars_all\n",
      "\n",
      "209368 MB RAM detected; reserving 104684 MB for main workspace.\n",
      "3 variants loaded from .bim file.\n",
      "646 people (363 males, 283 females) loaded from .fam.\n",
      "613 phenotype values loaded from .fam.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 646 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999484.\n",
      "--freq: Allele frequencies (founders only) written to\n",
      "/home/jupyter/workspace/freqs/AMR_path_vars_all.frq .\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/freqs/CAH_path_vars_all.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/workspace/bfiles/CAH_r7_pathogenic_only\n",
      "  --freq\n",
      "  --out /home/jupyter/workspace/freqs/CAH_path_vars_all\n",
      "\n",
      "209368 MB RAM detected; reserving 104684 MB for main workspace.\n",
      "2 variants loaded from .bim file.\n",
      "851 people (446 males, 405 females) loaded from .fam.\n",
      "827 phenotype values loaded from .fam.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 851 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "--freq: Allele frequencies (founders only) written to\n",
      "/home/jupyter/workspace/freqs/CAH_path_vars_all.frq .\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/freqs/CAS_path_vars_all.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/workspace/bfiles/CAS_r7_pathogenic_only\n",
      "  --freq\n",
      "  --out /home/jupyter/workspace/freqs/CAS_path_vars_all\n",
      "\n",
      "209368 MB RAM detected; reserving 104684 MB for main workspace.\n",
      "2 variants loaded from .bim file.\n",
      "903 people (439 males, 464 females) loaded from .fam.\n",
      "895 phenotype values loaded from .fam.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 903 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "--freq: Allele frequencies (founders only) written to\n",
      "/home/jupyter/workspace/freqs/CAS_path_vars_all.frq .\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/freqs/EAS_path_vars_all.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/workspace/bfiles/EAS_r7_pathogenic_only\n",
      "  --freq\n",
      "  --out /home/jupyter/workspace/freqs/EAS_path_vars_all\n",
      "\n",
      "209368 MB RAM detected; reserving 104684 MB for main workspace.\n",
      "5 variants loaded from .bim file.\n",
      "5167 people (3278 males, 1889 females) loaded from .fam.\n",
      "5123 phenotype values loaded from .fam.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 5167 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.998916.\n",
      "--freq: Allele frequencies (founders only) written to\n",
      "/home/jupyter/workspace/freqs/EAS_path_vars_all.frq .\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/freqs/EUR_path_vars_all.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/workspace/bfiles/EUR_r7_pathogenic_only\n",
      "  --freq\n",
      "  --out /home/jupyter/workspace/freqs/EUR_path_vars_all\n",
      "\n",
      "209368 MB RAM detected; reserving 104684 MB for main workspace.\n",
      "19 variants loaded from .bim file.\n",
      "38839 people (23296 males, 15543 females) loaded from .fam.\n",
      "30412 phenotype values loaded from .fam.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 38839 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.841842.\n",
      "--freq: Allele frequencies (founders only) written to\n",
      "/home/jupyter/workspace/freqs/EUR_path_vars_all.frq .\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/freqs/FIN_path_vars_all.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/workspace/bfiles/FIN_r7_pathogenic_only\n",
      "  --freq\n",
      "  --out /home/jupyter/workspace/freqs/FIN_path_vars_all\n",
      "\n",
      "209368 MB RAM detected; reserving 104684 MB for main workspace.\n",
      "1 variant loaded from .bim file.\n",
      "114 people (53 males, 61 females) loaded from .fam.\n",
      "106 phenotype values loaded from .fam.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 114 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.991228.\n",
      "--freq: Allele frequencies (founders only) written to\n",
      "/home/jupyter/workspace/freqs/FIN_path_vars_all.frq .\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/workspace/freqs/MDE_path_vars_all.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/workspace/bfiles/MDE_r7_pathogenic_only\n",
      "  --freq\n",
      "  --out /home/jupyter/workspace/freqs/MDE_path_vars_all\n",
      "\n",
      "209368 MB RAM detected; reserving 104684 MB for main workspace.\n",
      "3 variants loaded from .bim file.\n",
      "581 people (369 males, 212 females) loaded from .fam.\n",
      "536 phenotype values loaded from .fam.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 581 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999426.\n",
      "--freq: Allele frequencies (founders only) written to\n",
      "/home/jupyter/workspace/freqs/MDE_path_vars_all.frq .\n"
     ]
    }
   ],
   "source": [
    "# run all frequencies per ancestry\n",
    "\n",
    "for anc in ancestries:\n",
    "    anc_path = f'{workspace}/bfiles/{anc}_r7_pathogenic_only'\n",
    "    if os.path.isfile(f'{anc_path}.bim'):\n",
    "        plink_cmd = f'{plink_exec} --bfile {anc_path} --freq --out {workspace}/freqs/{anc}_path_vars_all'\n",
    "        !{plink_cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 90,
   "id": "3a058a4f-07bc-4b2c-8e18-48237e849c8c",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHR</th>\n",
       "      <th>SNP</th>\n",
       "      <th>A1</th>\n",
       "      <th>A2</th>\n",
       "      <th>MAF</th>\n",
       "      <th>MAF_A</th>\n",
       "      <th>MAF_U</th>\n",
       "      <th>MAF_AAC</th>\n",
       "      <th>MAF_A_AAC</th>\n",
       "      <th>MAF_U_AAC</th>\n",
       "      <th>...</th>\n",
       "      <th>MAF_U_EAS</th>\n",
       "      <th>MAF_EUR</th>\n",
       "      <th>MAF_A_EUR</th>\n",
       "      <th>MAF_U_EUR</th>\n",
       "      <th>MAF_FIN</th>\n",
       "      <th>MAF_A_FIN</th>\n",
       "      <th>MAF_U_FIN</th>\n",
       "      <th>MAF_MDE</th>\n",
       "      <th>MAF_A_MDE</th>\n",
       "      <th>MAF_U_MDE</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>chr1:155235196:G:A</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000348</td>\n",
       "      <td>0.000590</td>\n",
       "      <td>0.000109</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>chr1:155236295:G:A</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>chr1:155237453:C:T</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000206</td>\n",
       "      <td>0.000283</td>\n",
       "      <td>0.000054</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>chr1:155238206:A:C</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1</td>\n",
       "      <td>chr1:155238215:T:C</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>1</td>\n",
       "      <td>chr1:155238630:G:A</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000064</td>\n",
       "      <td>0.000094</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>1</td>\n",
       "      <td>chr1:155239934:G:A</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.001721</td>\n",
       "      <td>0.003215</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>1</td>\n",
       "      <td>chr1:16988455:C:T</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>1</td>\n",
       "      <td>chr1:16989961:G:A</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>1</td>\n",
       "      <td>chr1:16992345:G:A</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000026</td>\n",
       "      <td>0.000047</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>1</td>\n",
       "      <td>chr1:16996059:G:A</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000203</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>1</td>\n",
       "      <td>chr1:17005450:C:T</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>1</td>\n",
       "      <td>chr1:20645640:T:C</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000407</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>1</td>\n",
       "      <td>chr1:20649109:C:T</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000270</td>\n",
       "      <td>0.000378</td>\n",
       "      <td>0.000163</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000861</td>\n",
       "      <td>0.001608</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>1</td>\n",
       "      <td>chr1:20649217:C:T</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000407</td>\n",
       "      <td>0.000039</td>\n",
       "      <td>0.000024</td>\n",
       "      <td>0.000109</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>6</td>\n",
       "      <td>chr6:161350214:G:C</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>6</td>\n",
       "      <td>chr6:161785793:C:G</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000203</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>6</td>\n",
       "      <td>chr6:161785839:A:T</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>6</td>\n",
       "      <td>chr6:162443314:A:T</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>6</td>\n",
       "      <td>chr6:162443356:C:G</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000026</td>\n",
       "      <td>0.000047</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>12</td>\n",
       "      <td>chr12:40310434:C:T</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000219</td>\n",
       "      <td>0.000330</td>\n",
       "      <td>0.000054</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.003448</td>\n",
       "      <td>0.006452</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>15</td>\n",
       "      <td>chr15:89319225:T:A</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000039</td>\n",
       "      <td>0.000047</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>15</td>\n",
       "      <td>chr15:89321780:G:A</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000090</td>\n",
       "      <td>0.000094</td>\n",
       "      <td>0.000163</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>15</td>\n",
       "      <td>chr15:89321792:C:T</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000283</td>\n",
       "      <td>0.000283</td>\n",
       "      <td>0.000217</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>15</td>\n",
       "      <td>chr15:89325610:G:A</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>15</td>\n",
       "      <td>chr15:89325679:G:A</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000026</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000054</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>15</td>\n",
       "      <td>chr15:89327166:C:T</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>15</td>\n",
       "      <td>chr15:89327201:C:T</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.00045</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000624</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000736</td>\n",
       "      <td>0.000804</td>\n",
       "      <td>0.000544</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>15</td>\n",
       "      <td>chr15:89329041:G:A</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000026</td>\n",
       "      <td>0.000047</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>20</td>\n",
       "      <td>chr20:5100921:C:A</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>21</td>\n",
       "      <td>chr21:32726907:C:A</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>22</td>\n",
       "      <td>chr22:38115658:G:A</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000026</td>\n",
       "      <td>0.000047</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>22</td>\n",
       "      <td>chr22:38132917:C:A</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000203</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>22</td>\n",
       "      <td>chr22:38169318:G:A</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>34 rows Ã— 37 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    CHR                 SNP A1 A2  MAF MAF_A MAF_U  MAF_AAC  MAF_A_AAC  \\\n",
       "0     1  chr1:155235196:G:A  A  G  NaN   NaN   NaN      NaN        NaN   \n",
       "1     1  chr1:155236295:G:A  A  G  NaN   NaN   NaN      NaN        NaN   \n",
       "2     1  chr1:155237453:C:T  T  C  NaN   NaN   NaN      NaN        NaN   \n",
       "3     1  chr1:155238206:A:C  C  A  NaN   NaN   NaN      NaN        NaN   \n",
       "4     1  chr1:155238215:T:C  C  T  NaN   NaN   NaN      NaN        NaN   \n",
       "5     1  chr1:155238630:G:A  A  G  NaN   NaN   NaN      NaN        NaN   \n",
       "6     1  chr1:155239934:G:A  A  G  NaN   NaN   NaN      NaN        NaN   \n",
       "7     1   chr1:16988455:C:T  T  C  NaN   NaN   NaN      NaN        NaN   \n",
       "8     1   chr1:16989961:G:A  A  G  NaN   NaN   NaN      NaN        NaN   \n",
       "9     1   chr1:16992345:G:A  A  G  NaN   NaN   NaN      NaN        NaN   \n",
       "10    1   chr1:16996059:G:A  A  G  NaN   NaN   NaN      NaN        NaN   \n",
       "11    1   chr1:17005450:C:T  T  C  NaN   NaN   NaN      NaN        NaN   \n",
       "12    1   chr1:20645640:T:C  C  T  NaN   NaN   NaN      NaN        NaN   \n",
       "13    1   chr1:20649109:C:T  T  C  NaN   NaN   NaN      NaN        NaN   \n",
       "14    1   chr1:20649217:C:T  T  C  NaN   NaN   NaN      NaN        NaN   \n",
       "15    6  chr6:161350214:G:C  C  G  NaN   NaN   NaN      NaN        NaN   \n",
       "16    6  chr6:161785793:C:G  G  C  NaN   NaN   NaN      NaN        NaN   \n",
       "17    6  chr6:161785839:A:T  T  A  NaN   NaN   NaN      NaN        NaN   \n",
       "18    6  chr6:162443314:A:T  T  A  NaN   NaN   NaN      NaN        NaN   \n",
       "19    6  chr6:162443356:C:G  G  C  NaN   NaN   NaN      NaN        NaN   \n",
       "20   12  chr12:40310434:C:T  T  C  NaN   NaN   NaN      NaN        NaN   \n",
       "21   15  chr15:89319225:T:A  A  T  NaN   NaN   NaN      NaN        NaN   \n",
       "22   15  chr15:89321780:G:A  A  G  NaN   NaN   NaN      NaN        NaN   \n",
       "23   15  chr15:89321792:C:T  T  C  NaN   NaN   NaN  0.00000        0.0   \n",
       "24   15  chr15:89325610:G:A  A  G  NaN   NaN   NaN      NaN        NaN   \n",
       "25   15  chr15:89325679:G:A  A  G  NaN   NaN   NaN      NaN        NaN   \n",
       "26   15  chr15:89327166:C:T  T  C  NaN   NaN   NaN      NaN        NaN   \n",
       "27   15  chr15:89327201:C:T  T  C  NaN   NaN   NaN  0.00045        0.0   \n",
       "28   15  chr15:89329041:G:A  A  G  NaN   NaN   NaN      NaN        NaN   \n",
       "29   20   chr20:5100921:C:A  A  C  NaN   NaN   NaN      NaN        NaN   \n",
       "30   21  chr21:32726907:C:A  A  C  NaN   NaN   NaN      NaN        NaN   \n",
       "31   22  chr22:38115658:G:A  A  G  NaN   NaN   NaN      NaN        NaN   \n",
       "32   22  chr22:38132917:C:A  A  C  NaN   NaN   NaN      NaN        NaN   \n",
       "33   22  chr22:38169318:G:A  A  G  NaN   NaN   NaN      NaN        NaN   \n",
       "\n",
       "    MAF_U_AAC  ...  MAF_U_EAS   MAF_EUR  MAF_A_EUR  MAF_U_EUR  MAF_FIN  \\\n",
       "0         NaN  ...        NaN  0.000348   0.000590   0.000109      NaN   \n",
       "1         NaN  ...        NaN       NaN        NaN        NaN      NaN   \n",
       "2         NaN  ...        NaN  0.000206   0.000283   0.000054      NaN   \n",
       "3         NaN  ...        NaN       NaN        NaN        NaN      NaN   \n",
       "4         NaN  ...        NaN  0.000000   0.000000   0.000000      NaN   \n",
       "5         NaN  ...        NaN  0.000064   0.000094   0.000000      NaN   \n",
       "6         NaN  ...        NaN       NaN        NaN        NaN      NaN   \n",
       "7         NaN  ...        NaN       NaN        NaN        NaN      NaN   \n",
       "8         NaN  ...        NaN       NaN        NaN        NaN      NaN   \n",
       "9         NaN  ...        NaN  0.000026   0.000047   0.000000      NaN   \n",
       "10        NaN  ...   0.000203  0.000000   0.000000   0.000000      NaN   \n",
       "11        NaN  ...        NaN  0.000000   0.000000   0.000000      NaN   \n",
       "12        NaN  ...   0.000407       NaN        NaN        NaN      NaN   \n",
       "13        NaN  ...        NaN  0.000270   0.000378   0.000163      NaN   \n",
       "14        NaN  ...   0.000407  0.000039   0.000024   0.000109      0.0   \n",
       "15        NaN  ...        NaN       NaN        NaN        NaN      NaN   \n",
       "16        NaN  ...   0.000203       NaN        NaN        NaN      NaN   \n",
       "17        NaN  ...        NaN       NaN        NaN        NaN      NaN   \n",
       "18        NaN  ...        NaN       NaN        NaN        NaN      NaN   \n",
       "19        NaN  ...        NaN  0.000026   0.000047   0.000000      NaN   \n",
       "20        NaN  ...        NaN  0.000219   0.000330   0.000054      NaN   \n",
       "21        NaN  ...        NaN  0.000039   0.000047   0.000000      NaN   \n",
       "22        NaN  ...        NaN  0.000090   0.000094   0.000163      NaN   \n",
       "23   0.000000  ...        NaN  0.000283   0.000283   0.000217      NaN   \n",
       "24        NaN  ...        NaN       NaN        NaN        NaN      NaN   \n",
       "25        NaN  ...        NaN  0.000026   0.000000   0.000054      NaN   \n",
       "26        NaN  ...        NaN       NaN        NaN        NaN      NaN   \n",
       "27   0.000624  ...        NaN  0.000736   0.000804   0.000544      NaN   \n",
       "28        NaN  ...        NaN  0.000026   0.000047   0.000000      NaN   \n",
       "29        NaN  ...        NaN  0.000000   0.000000   0.000000      NaN   \n",
       "30        NaN  ...        NaN       NaN        NaN        NaN      NaN   \n",
       "31        NaN  ...        NaN  0.000026   0.000047   0.000000      NaN   \n",
       "32        NaN  ...   0.000203       NaN        NaN        NaN      NaN   \n",
       "33        NaN  ...        NaN       NaN        NaN        NaN      NaN   \n",
       "\n",
       "    MAF_A_FIN  MAF_U_FIN   MAF_MDE  MAF_A_MDE  MAF_U_MDE  \n",
       "0         NaN        NaN       NaN        NaN        NaN  \n",
       "1         NaN        NaN       NaN        NaN        NaN  \n",
       "2         NaN        NaN       NaN        NaN        NaN  \n",
       "3         NaN        NaN       NaN        NaN        NaN  \n",
       "4         NaN        NaN       NaN        NaN        NaN  \n",
       "5         NaN        NaN       NaN        NaN        NaN  \n",
       "6         NaN        NaN  0.001721   0.003215        0.0  \n",
       "7         NaN        NaN       NaN        NaN        NaN  \n",
       "8         NaN        NaN       NaN        NaN        NaN  \n",
       "9         NaN        NaN       NaN        NaN        NaN  \n",
       "10        NaN        NaN       NaN        NaN        NaN  \n",
       "11        NaN        NaN       NaN        NaN        NaN  \n",
       "12        NaN        NaN       NaN        NaN        NaN  \n",
       "13        NaN        NaN  0.000861   0.001608        0.0  \n",
       "14        0.0        0.0       NaN        NaN        NaN  \n",
       "15        NaN        NaN       NaN        NaN        NaN  \n",
       "16        NaN        NaN       NaN        NaN        NaN  \n",
       "17        NaN        NaN       NaN        NaN        NaN  \n",
       "18        NaN        NaN       NaN        NaN        NaN  \n",
       "19        NaN        NaN       NaN        NaN        NaN  \n",
       "20        NaN        NaN  0.003448   0.006452        0.0  \n",
       "21        NaN        NaN       NaN        NaN        NaN  \n",
       "22        NaN        NaN       NaN        NaN        NaN  \n",
       "23        NaN        NaN       NaN        NaN        NaN  \n",
       "24        NaN        NaN       NaN        NaN        NaN  \n",
       "25        NaN        NaN       NaN        NaN        NaN  \n",
       "26        NaN        NaN       NaN        NaN        NaN  \n",
       "27        NaN        NaN       NaN        NaN        NaN  \n",
       "28        NaN        NaN       NaN        NaN        NaN  \n",
       "29        NaN        NaN       NaN        NaN        NaN  \n",
       "30        NaN        NaN       NaN        NaN        NaN  \n",
       "31        NaN        NaN       NaN        NaN        NaN  \n",
       "32        NaN        NaN       NaN        NaN        NaN  \n",
       "33        NaN        NaN       NaN        NaN        NaN  \n",
       "\n",
       "[34 rows x 37 columns]"
      ]
     },
     "execution_count": 90,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# combine freq calculations\n",
    "\n",
    "# SAS has no pathogenic vars\n",
    "all_anc_freqs = pd.DataFrame(columns=['CHR','SNP','A1','A2','MAF','MAF_A','MAF_U'])\n",
    "\n",
    "for anc in ancestries:\n",
    "    if os.path.isfile(f'{workspace}/bfiles/{anc}_r7_pathogenic_only.bim'):\n",
    "        all_freqs = pd.read_csv(f'{workspace}/freqs/{anc}_path_vars_all.frq', sep='\\s+', usecols=['CHR','SNP','A1','A2','MAF'])\n",
    "        case_control_freqs = pd.read_csv(f'{workspace}/freqs/{anc}_path_vars_case_control.frq.cc', sep='\\s+', usecols=['CHR','SNP','A1','A2','MAF_A','MAF_U'])\n",
    "        freqs = pd.merge(all_freqs, case_control_freqs, on=['CHR','SNP','A1','A2'], how='outer')\n",
    "        all_anc_freqs = all_anc_freqs.merge(freqs, on=['CHR','SNP','A1','A2'], how='outer', suffixes=(None, f'_{anc}'))\n",
    "\n",
    "all_anc_freqs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 95,
   "id": "b7a73dc7-e938-4db3-aab5-f66ac2edd6b6",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>GENE</th>\n",
       "      <th>snpID</th>\n",
       "      <th>MAF_AAC</th>\n",
       "      <th>MAF_AFR</th>\n",
       "      <th>MAF_AJ</th>\n",
       "      <th>MAF_AMR</th>\n",
       "      <th>MAF_CAH</th>\n",
       "      <th>MAF_CAS</th>\n",
       "      <th>MAF_EAS</th>\n",
       "      <th>MAF_EUR</th>\n",
       "      <th>MAF_FIN</th>\n",
       "      <th>MAF_MDE</th>\n",
       "      <th>MAF_SAS</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ATP13A2</td>\n",
       "      <td>5</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GBA</td>\n",
       "      <td>7</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>LRRK2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>PINK1</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>PLA2G6</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>POLG</td>\n",
       "      <td>8</td>\n",
       "      <td>2</td>\n",
       "      <td>4</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>6</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>PRKN</td>\n",
       "      <td>5</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>SYNJ1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>TMEM230</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      GENE  snpID  MAF_AAC  MAF_AFR  MAF_AJ  MAF_AMR  MAF_CAH  MAF_CAS  \\\n",
       "0  ATP13A2      5        0        1       0        1        0        0   \n",
       "1      GBA      7        0        2       0        0        1        0   \n",
       "2    LRRK2      1        0        0       0        0        0        1   \n",
       "3    PINK1      3        0        0       0        0        0        0   \n",
       "4   PLA2G6      3        0        0       0        1        0        1   \n",
       "5     POLG      8        2        4       1        0        0        0   \n",
       "6     PRKN      5        0        0       1        1        1        0   \n",
       "7    SYNJ1      1        0        1       0        0        0        0   \n",
       "8  TMEM230      1        0        0       0        0        0        0   \n",
       "\n",
       "   MAF_EAS  MAF_EUR  MAF_FIN  MAF_MDE  MAF_SAS  \n",
       "0        1        3        0        0        0  \n",
       "1        0        4        0        1        0  \n",
       "2        0        1        0        1        0  \n",
       "3        2        2        1        1        0  \n",
       "4        1        1        0        0        0  \n",
       "5        0        6        0        0        0  \n",
       "6        1        1        0        0        0  \n",
       "7        0        0        0        0        0  \n",
       "8        0        1        0        0        0  "
      ]
     },
     "execution_count": 95,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# get number of pathogenic variants within each PD gene of interest in each ancestry\n",
    "\n",
    "gene_path_vars = annotated_pathogenic_vars[['Gene.refGene','avsnp151','snpID']].copy()\n",
    "gene_path_vars = gene_path_vars.merge(all_anc_freqs, left_on=['snpID'], right_on=['SNP'], how='outer')\n",
    "gene_path_vars = gene_path_vars.rename(columns={'Gene.refGene':'GENE', 'MAF':'MAF_SAS', 'MAF_A':'MAF_A_SAS', 'MAF_U':'MAF_U_SAS'})\n",
    "\n",
    "gene_path_vars.fillna('NA').to_csv(f'{workspace}/all_anc_freqs.txt', sep='\\t', header=True, index=False)\n",
    "\n",
    "drop_cols = ['avsnp151','CHR','SNP','A1','A2'] + [f'MAF_A_{anc}' for anc in ancestries] + [f'MAF_U_{anc}' for anc in ancestries]\n",
    "per_gene_path_vars = gene_path_vars.drop(drop_cols, axis=1)\n",
    "col_order = ['GENE','snpID'] + [f'MAF_{anc}' for anc in ancestries]\n",
    "per_gene_path_vars = per_gene_path_vars[col_order]\n",
    "\n",
    "per_gene_counts = per_gene_path_vars.groupby('GENE').count().reset_index()\n",
    "per_gene_counts"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 92,
   "id": "c9a0a011-fed0-444c-9a77-688cdb397f24",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# blauwendraat PD gene classifications\n",
    "\n",
    "gene_classifications = {'SNCA':'very high',\\\n",
    "                        'PRKN':'very high',\\\n",
    "                        'UCHL1':'low',\\\n",
    "                        'PARK7':'very high',\\\n",
    "                        'LRRK2':'very high',\\\n",
    "                        'PINK1':'very high',\\\n",
    "                        'POLG':'high',\\\n",
    "                        'HTRA2':'low',\\\n",
    "                        'ATP13A2':'very high',\\\n",
    "                        'FBXO7':'very high',\\\n",
    "                        'GIGYF2':'low',\\\n",
    "                        'GBA':'very high',\\\n",
    "                        'PLA2G6':'very high',\\\n",
    "                        'EIF4G1':'low',\\\n",
    "                        'VPS35':'very high',\\\n",
    "                        'DNAJC6':'high',\\\n",
    "                        'SYNJ1':'high',\\\n",
    "                        'DNAJC13':'low',\\\n",
    "                        'TMEM230':'low',\\\n",
    "                        'VPS13C':'high',\\\n",
    "                        'LRP10':'low'}\n",
    "\n",
    "gene_classifications = pd.DataFrame.from_dict(gene_classifications, orient='index').reset_index().rename(columns={'index':'GENE',0:'class'})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 94,
   "id": "831245b8-4fed-40ab-9bbe-b7611165461d",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# count how many pathogenic variants per gene are present in each ancestry\n",
    "\n",
    "col_order = ['GENE','class','snpID'] + [f'MAF_{anc}' for anc in ancestries]\n",
    "per_gene_class_counts = per_gene_counts.merge(gene_classifications, on='GENE', how='outer')[col_order]\n",
    "\n",
    "class_ord = {'low':2,'high':1,'very high':0}\n",
    "per_gene_class_counts.sort_values(by='class', key=lambda x: x.map(class_ord)).fillna(0).to_csv(f'{workspace}/per_gene_per_anc_pathogenic_var_counts.txt', sep='\\t', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "00cb322f-044a-405d-8158-23197d7e48f4",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "206a6c9c-6a7b-4acd-a7af-ad4f4c92ad03",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene</th>\n",
       "      <th>gnomAD_ID</th>\n",
       "      <th>Chromosome</th>\n",
       "      <th>Position</th>\n",
       "      <th>rsIDs</th>\n",
       "      <th>Reference</th>\n",
       "      <th>Alternate</th>\n",
       "      <th>ClinVar_ClnSignificance</th>\n",
       "      <th>ClinVar_varID</th>\n",
       "      <th>allele_count</th>\n",
       "      <th>allele_number</th>\n",
       "      <th>allele_freq</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ATP13A2</td>\n",
       "      <td>1-16986881-CAG-C</td>\n",
       "      <td>1</td>\n",
       "      <td>16986881</td>\n",
       "      <td>rs1570759415</td>\n",
       "      <td>CAG</td>\n",
       "      <td>C</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>660925.0</td>\n",
       "      <td>1</td>\n",
       "      <td>1614048</td>\n",
       "      <td>6.195603e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ATP13A2</td>\n",
       "      <td>1-16986885-GAGA-G</td>\n",
       "      <td>1</td>\n",
       "      <td>16986885</td>\n",
       "      <td>rs1057519290</td>\n",
       "      <td>GAGA</td>\n",
       "      <td>G</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>374887.0</td>\n",
       "      <td>1</td>\n",
       "      <td>1614104</td>\n",
       "      <td>6.195388e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ATP13A2</td>\n",
       "      <td>1-16986886-A-AG</td>\n",
       "      <td>1</td>\n",
       "      <td>16986886</td>\n",
       "      <td>rs747617559</td>\n",
       "      <td>A</td>\n",
       "      <td>AG</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>1968613.0</td>\n",
       "      <td>2</td>\n",
       "      <td>1614040</td>\n",
       "      <td>1.239127e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ATP13A2</td>\n",
       "      <td>1-16988455-C-T</td>\n",
       "      <td>1</td>\n",
       "      <td>16988455</td>\n",
       "      <td>rs144701072</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>66099.0</td>\n",
       "      <td>15</td>\n",
       "      <td>1613774</td>\n",
       "      <td>9.294982e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ATP13A2</td>\n",
       "      <td>1-16989739-A-C</td>\n",
       "      <td>1</td>\n",
       "      <td>16989739</td>\n",
       "      <td>rs587777053</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>66098.0</td>\n",
       "      <td>1</td>\n",
       "      <td>1614078</td>\n",
       "      <td>6.195487e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>235</th>\n",
       "      <td>TMEM230</td>\n",
       "      <td>20-5100921-C-A</td>\n",
       "      <td>20</td>\n",
       "      <td>5100921</td>\n",
       "      <td>rs764786986</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>243014.0</td>\n",
       "      <td>5</td>\n",
       "      <td>1613958</td>\n",
       "      <td>3.097974e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>236</th>\n",
       "      <td>UCHL1</td>\n",
       "      <td>4-41256996-A-C</td>\n",
       "      <td>4</td>\n",
       "      <td>41256996</td>\n",
       "      <td>rs397515634</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>88635.0</td>\n",
       "      <td>2</td>\n",
       "      <td>1614182</td>\n",
       "      <td>1.239018e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>237</th>\n",
       "      <td>UCHL1</td>\n",
       "      <td>4-41257706-C-CGCT</td>\n",
       "      <td>4</td>\n",
       "      <td>41257706</td>\n",
       "      <td>rs749368841</td>\n",
       "      <td>C</td>\n",
       "      <td>CGCT</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>2077519.0</td>\n",
       "      <td>3</td>\n",
       "      <td>1580822</td>\n",
       "      <td>1.897747e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>238</th>\n",
       "      <td>UCHL1</td>\n",
       "      <td>4-41261751-CAG-C</td>\n",
       "      <td>4</td>\n",
       "      <td>41261751</td>\n",
       "      <td>rs1310363710</td>\n",
       "      <td>CAG</td>\n",
       "      <td>C</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>3068443.0</td>\n",
       "      <td>1</td>\n",
       "      <td>1613326</td>\n",
       "      <td>6.198375e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>239</th>\n",
       "      <td>UCHL1</td>\n",
       "      <td>4-41264108-C-T</td>\n",
       "      <td>4</td>\n",
       "      <td>41264108</td>\n",
       "      <td>rs2154087267</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>1380167.0</td>\n",
       "      <td>1</td>\n",
       "      <td>1614208</td>\n",
       "      <td>6.194989e-07</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>240 rows Ã— 12 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        gene          gnomAD_ID  Chromosome  Position         rsIDs Reference  \\\n",
       "0    ATP13A2   1-16986881-CAG-C           1  16986881  rs1570759415       CAG   \n",
       "1    ATP13A2  1-16986885-GAGA-G           1  16986885  rs1057519290      GAGA   \n",
       "2    ATP13A2    1-16986886-A-AG           1  16986886   rs747617559         A   \n",
       "3    ATP13A2     1-16988455-C-T           1  16988455   rs144701072         C   \n",
       "4    ATP13A2     1-16989739-A-C           1  16989739   rs587777053         A   \n",
       "..       ...                ...         ...       ...           ...       ...   \n",
       "235  TMEM230     20-5100921-C-A          20   5100921   rs764786986         C   \n",
       "236    UCHL1     4-41256996-A-C           4  41256996   rs397515634         A   \n",
       "237    UCHL1  4-41257706-C-CGCT           4  41257706   rs749368841         C   \n",
       "238    UCHL1   4-41261751-CAG-C           4  41261751  rs1310363710       CAG   \n",
       "239    UCHL1     4-41264108-C-T           4  41264108  rs2154087267         C   \n",
       "\n",
       "    Alternate ClinVar_ClnSignificance  ClinVar_varID  allele_count  \\\n",
       "0           C              Pathogenic       660925.0             1   \n",
       "1           G              Pathogenic       374887.0             1   \n",
       "2          AG              Pathogenic      1968613.0             2   \n",
       "3           T              Pathogenic        66099.0            15   \n",
       "4           C              Pathogenic        66098.0             1   \n",
       "..        ...                     ...            ...           ...   \n",
       "235         A              Pathogenic       243014.0             5   \n",
       "236         C              Pathogenic        88635.0             2   \n",
       "237      CGCT              Pathogenic      2077519.0             3   \n",
       "238         C              Pathogenic      3068443.0             1   \n",
       "239         T              Pathogenic      1380167.0             1   \n",
       "\n",
       "     allele_number   allele_freq  \n",
       "0          1614048  6.195603e-07  \n",
       "1          1614104  6.195388e-07  \n",
       "2          1614040  1.239127e-06  \n",
       "3          1613774  9.294982e-06  \n",
       "4          1614078  6.195487e-07  \n",
       "..             ...           ...  \n",
       "235        1613958  3.097974e-06  \n",
       "236        1614182  1.239018e-06  \n",
       "237        1580822  1.897747e-06  \n",
       "238        1613326  6.198375e-07  \n",
       "239        1614208  6.194989e-07  \n",
       "\n",
       "[240 rows x 12 columns]"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# compare pathogenic variants with gnomAD annotations\n",
    "# - annotations were pulled manually for each gene of interest from gnomAD site and then combined to one file\n",
    "\n",
    "gnomad_info = pd.read_csv(f'{workspace}/gnomAD_selectedgenes_pathogenic_vars.txt', sep='\\t')\n",
    "gnomad_info"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "6dc1c354-9c88-403d-93eb-02a0eed29f53",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "annotated_path_vars = pd.read_csv(f'{workspace}/annotated_pathogenic_vars.txt', sep='\\t')\n",
    "\n",
    "gnomad_info['in_gp2'] = gnomad_info['rsIDs'].isin(annotated_path_vars['avsnp151'].tolist())\n",
    "gnomad_info.to_csv(f'{workspace}/gnomAD_inGP2.txt', sep='\\t', header=True, index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4c19139d-c6d1-456a-a647-2ac4d4a391bb",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "environment": {
   "kernel": "python3",
   "name": "r-cpu.m125",
   "type": "gcloud",
   "uri": "us-docker.pkg.dev/deeplearning-platform-release/gcr.io/r-cpu:m125"
  },
  "kernelspec": {
   "display_name": "Python 3 (Local)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.15"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
